From 5db3c4669a87849adf51fcf84acca7a1b83ab169 Mon Sep 17 00:00:00 2001
From: Michael Cook <code@mdcook.net>
Date: Tue, 2 Jan 2018 11:09:00 +0100
Subject: [PATCH 058/249] wiki: bluetooth device schema validation

- convert bluetooth to bluetooth.profiles (optional)
  and bluetooth.spec (required)
- display in wiki as bluetooth.spec with bluetooth.profiles
- set bluetooth to required in schema
- prevent a couple unrelated cases that allowed empty arrays

Change-Id: Icaad595a87a26e407e6111d6da9f6ead9858318f
---
 _data/devices/A6020.yml         |  2 +-
 _data/devices/FP2.yml           |  2 +-
 _data/devices/P024.yml          |  2 +-
 _data/devices/Z008.yml          |  2 +-
 _data/devices/Z00A.yml          |  2 +-
 _data/devices/Z00D.yml          |  2 +-
 _data/devices/Z00L.yml          |  2 +-
 _data/devices/Z00T.yml          |  2 +-
 _data/devices/addison.yml       |  2 +-
 _data/devices/angler.yml        |  2 +-
 _data/devices/armani.yml        |  2 +-
 _data/devices/athene.yml        |  2 +-
 _data/devices/axon7.yml         |  2 +-
 _data/devices/bacon.yml         |  2 +-
 _data/devices/bullhead.yml      |  2 +-
 _data/devices/cancro.yml        |  2 +-
 _data/devices/capricorn.yml     |  2 +-
 _data/devices/chagallwifi.yml   |  2 +-
 _data/devices/che10.yml         |  2 +-
 _data/devices/cheeseburger.yml  |  2 +-
 _data/devices/cherry.yml        |  2 +-
 _data/devices/clark.yml         |  2 +-
 _data/devices/condor.yml        |  2 +-
 _data/devices/crackling.yml     |  2 +-
 _data/devices/d2att.yml         |  2 +-
 _data/devices/d2spr.yml         |  2 +-
 _data/devices/d2tmo.yml         |  2 +-
 _data/devices/d2vzw.yml         |  2 +-
 _data/devices/d800.yml          |  2 +-
 _data/devices/d801.yml          |  2 +-
 _data/devices/d802.yml          |  2 +-
 _data/devices/d803.yml          |  2 +-
 _data/devices/d850.yml          |  2 +-
 _data/devices/d851.yml          |  2 +-
 _data/devices/d852.yml          |  2 +-
 _data/devices/d855.yml          |  2 +-
 _data/devices/deb.yml           |  2 +-
 _data/devices/dogo.yml          |  2 +-
 _data/devices/dragon.yml        |  2 +-
 _data/devices/espresso3g.yml    |  2 +-
 _data/devices/espressowifi.yml  |  2 +-
 _data/devices/ether.yml         |  2 +-
 _data/devices/f400.yml          |  2 +-
 _data/devices/falcon.yml        |  2 +-
 _data/devices/find7.yml         |  2 +-
 _data/devices/find7s.yml        |  2 +-
 _data/devices/flo.yml           |  2 +-
 _data/devices/flounder.yml      |  2 +-
 _data/devices/flounder_lte.yml  |  2 +-
 _data/devices/foster.yml        |  2 +-
 _data/devices/fugu.yml          |  2 +-
 _data/devices/g2m.yml           |  2 +-
 _data/devices/gemini.yml        |  2 +-
 _data/devices/ghost.yml         |  2 +-
 _data/devices/gohan.yml         |  2 +-
 _data/devices/griffin.yml       |  2 +-
 _data/devices/gtelwifiue.yml    |  2 +-
 _data/devices/gtesqltespr.yml   |  2 +-
 _data/devices/gts210ltexx.yml   |  2 +-
 _data/devices/gts210vewifi.yml  |  2 +-
 _data/devices/gts210wifi.yml    |  2 +-
 _data/devices/gts28vewifi.yml   |  2 +-
 _data/devices/h811.yml          |  2 +-
 _data/devices/h815.yml          |  2 +-
 _data/devices/h830.yml          |  2 +-
 _data/devices/h850.yml          |  2 +-
 _data/devices/h870.yml          |  2 +-
 _data/devices/h910.yml          |  2 +-
 _data/devices/h918.yml          |  2 +-
 _data/devices/ha3g.yml          |  2 +-
 _data/devices/ham.yml           |  2 +-
 _data/devices/hammerhead.yml    |  2 +-
 _data/devices/harpia.yml        |  2 +-
 _data/devices/hayabusa.yml      |  2 +-
 _data/devices/hero2lte.yml      |  2 +-
 _data/devices/herolte.yml       |  2 +-
 _data/devices/hiaeuhl.yml       |  2 +-
 _data/devices/hiaeul.yml        |  2 +-
 _data/devices/himaul.yml        |  2 +-
 _data/devices/himawl.yml        |  2 +-
 _data/devices/hlte.yml          |  2 +-
 _data/devices/hltetmo.yml       |  2 +-
 _data/devices/huashan.yml       |  2 +-
 _data/devices/hydrogen.yml      |  2 +-
 _data/devices/i9100.yml         |  2 +-
 _data/devices/i9300.yml         |  2 +-
 _data/devices/i9305.yml         |  2 +-
 _data/devices/ido.yml           |  2 +-
 _data/devices/ivy.yml           |  2 +-
 _data/devices/jag3gds.yml       |  2 +-
 _data/devices/jagnm.yml         |  2 +-
 _data/devices/jalebi.yml        |  2 +-
 _data/devices/jfltevzw.yml      |  2 +-
 _data/devices/jfltexx.yml       |  2 +-
 _data/devices/k3gxx.yml         |  2 +-
 _data/devices/karin.yml         |  2 +-
 _data/devices/karin_windy.yml   |  2 +-
 _data/devices/kccat6.yml        |  2 +-
 _data/devices/kenzo.yml         |  2 +-
 _data/devices/kingdom.yml       |  2 +-
 _data/devices/kipper.yml        |  2 +-
 _data/devices/kiwi.yml          |  2 +-
 _data/devices/klimtwifi.yml     |  2 +-
 _data/devices/klte.yml          |  2 +-
 _data/devices/kltechn.yml       |  2 +-
 _data/devices/kltechnduo.yml    |  2 +-
 _data/devices/klteduos.yml      |  2 +-
 _data/devices/kltedv.yml        |  2 +-
 _data/devices/kltekdi.yml       |  2 +-
 _data/devices/kltekor.yml       |  2 +-
 _data/devices/kltesprsports.yml |  2 +-
 _data/devices/ks01lte.yml       |  2 +-
 _data/devices/lentislte.yml     |  2 +-
 _data/devices/lettuce.yml       |  2 +-
 _data/devices/libra.yml         |  2 +-
 _data/devices/lithium.yml       |  2 +-
 _data/devices/ls990.yml         |  2 +-
 _data/devices/ls997.yml         |  2 +-
 _data/devices/lt02ltespr.yml    |  2 +-
 _data/devices/lt03lte.yml       |  2 +-
 _data/devices/lux.yml           |  2 +-
 _data/devices/m216.yml          |  2 +-
 _data/devices/m7.yml            |  2 +-
 _data/devices/m7vzw.yml         |  2 +-
 _data/devices/m8.yml            |  2 +-
 _data/devices/m8d.yml           |  2 +-
 _data/devices/maguro.yml        |  2 +-
 _data/devices/mako.yml          |  2 +-
 _data/devices/manta.yml         |  2 +-
 _data/devices/maserati.yml      |  2 +-
 _data/devices/merlin.yml        |  2 +-
 _data/devices/mido.yml          |  2 +-
 _data/devices/mint.yml          |  2 +-
 _data/devices/mondrianwifi.yml  |  2 +-
 _data/devices/mt2.yml           |  2 +-
 _data/devices/n1awifi.yml       |  2 +-
 _data/devices/n2awifi.yml       |  2 +-
 _data/devices/n3.yml            |  2 +-
 _data/devices/n5100.yml         |  2 +-
 _data/devices/n5110.yml         |  2 +-
 _data/devices/n5120.yml         |  2 +-
 _data/devices/natrium.yml       |  2 +-
 _data/devices/nicki.yml         |  2 +-
 _data/devices/nx512j.yml        |  2 +-
 _data/devices/odin.yml          |  2 +-
 _data/devices/oneplus2.yml      |  2 +-
 _data/devices/oneplus3.yml      |  2 +-
 _data/devices/onyx.yml          |  2 +-
 _data/devices/osprey.yml        |  2 +-
 _data/devices/otus.yml          |  2 +-
 _data/devices/paella.yml        |  2 +-
 _data/devices/peach.yml         |  2 +-
 _data/devices/peregrine.yml     |  2 +-
 _data/devices/piccolo.yml       |  2 +-
 _data/devices/pme.yml           |  2 +-
 _data/devices/pollux.yml        |  2 +-
 _data/devices/pollux_windy.yml  |  2 +-
 _data/devices/r5.yml            |  2 +-
 _data/devices/r7plus.yml        |  2 +-
 _data/devices/r7sf.yml          |  2 +-
 _data/devices/roth.yml          |  2 +-
 _data/devices/s2.yml            |  2 +-
 _data/devices/seed.yml          |  2 +-
 _data/devices/serrano3gxx.yml   |  2 +-
 _data/devices/serranodsdd.yml   |  2 +-
 _data/devices/serranoltexx.yml  |  2 +-
 _data/devices/shamu.yml         |  2 +-
 _data/devices/shieldtablet.yml  |  2 +-
 _data/devices/spyder.yml        |  2 +-
 _data/devices/surnia.yml        |  2 +-
 _data/devices/suzuran.yml       |  2 +-
 _data/devices/t0lte.yml         |  2 +-
 _data/devices/t6.yml            |  2 +-
 _data/devices/t6vzw.yml         |  2 +-
 _data/devices/taoshan.yml       |  2 +-
 _data/devices/targa.yml         |  2 +-
 _data/devices/tenshi.yml        |  2 +-
 _data/devices/thea.yml          |  2 +-
 _data/devices/titan.yml         |  2 +-
 _data/devices/tomato.yml        |  2 +-
 _data/devices/toro.yml          |  2 +-
 _data/devices/toroplus.yml      |  2 +-
 _data/devices/tsubasa.yml       |  2 +-
 _data/devices/umts_spyder.yml   |  2 +-
 _data/devices/us996.yml         |  2 +-
 _data/devices/us997.yml         |  2 +-
 _data/devices/v1awifi.yml       |  2 +-
 _data/devices/v400.yml          |  2 +-
 _data/devices/v410.yml          |  2 +-
 _data/devices/v480.yml          |  2 +-
 _data/devices/v500.yml          |  2 +-
 _data/devices/v521.yml          |  2 +-
 _data/devices/vegetalte.yml     |  2 +-
 _data/devices/victara.yml       |  2 +-
 _data/devices/vs985.yml         |  2 +-
 _data/devices/vs995.yml         |  2 +-
 _data/devices/w5.yml            |  2 +-
 _data/devices/w7.yml            |  2 +-
 _data/devices/wt88047.yml       |  2 +-
 _data/devices/x2.yml            |  2 +-
 _data/devices/xt897.yml         |  2 +-
 _data/devices/yuga.yml          |  2 +-
 _data/devices/zl1.yml           |  2 +-
 _includes/deviceinfo.html       |  8 +++++++-
 device_sample/sample.yml        |  2 +-
 test/schema-06.yml              | 36 +++++++++++++++++++++++++++++++++
 206 files changed, 247 insertions(+), 205 deletions(-)

diff --git a/_data/devices/A6020.yml b/_data/devices/A6020.yml
index 3b3c975..b5771be 100644
--- a/_data/devices/A6020.yml
+++ b/_data/devices/A6020.yml
@@ -1,6 +1,6 @@
 architecture: arm64
 battery: Removable Li-Ion 2750 mAh
-bluetooth: v4.1 with A2DP, LE
+bluetooth: {spec: '4.1', profiles: [A2DP]}
 cameras:
 - {flash: 'LED', info: '13 MP'}
 - {flash: '', info: '5 MP'}
diff --git a/_data/devices/FP2.yml b/_data/devices/FP2.yml
index 4db59ea..6d9acb4 100644
--- a/_data/devices/FP2.yml
+++ b/_data/devices/FP2.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Removable Li-Ion 2420 mAh
-bluetooth: 4.0 BLE
+bluetooth: {spec: 4.0 + LE}
 cameras:
 - {flash: 'LED', info: '12 MP'}
 - {flash: '', info: '5 MP'}
diff --git a/_data/devices/P024.yml b/_data/devices/P024.yml
index ab6edef..b106633 100644
--- a/_data/devices/P024.yml
+++ b/_data/devices/P024.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Non-removable Li-Po 4000 mAh
-bluetooth: v4.0 with A2DP, LE
+bluetooth: {spec: 4.0 + LE, profiles: [A2DP]}
 cameras:
 - {flash: '', info: '5 / 8 MP'}
 - {flash: '', info: '2 MP'}
diff --git a/_data/devices/Z008.yml b/_data/devices/Z008.yml
index cc5c145..9fea2dd 100644
--- a/_data/devices/Z008.yml
+++ b/_data/devices/Z008.yml
@@ -1,6 +1,6 @@
 architecture: x86
 battery: 3000mAh
-bluetooth: v4.0, A2DP, EDR
+bluetooth: {spec: '4.0', profiles: [A2DP]}
 cameras:
 - {flash: 'LED', info: '13 MP'}
 - {flash: '', info: '5 MP'}
diff --git a/_data/devices/Z00A.yml b/_data/devices/Z00A.yml
index 25546a8..566a2a5 100644
--- a/_data/devices/Z00A.yml
+++ b/_data/devices/Z00A.yml
@@ -1,6 +1,6 @@
 architecture: x86
 battery: 3000mAh
-bluetooth: v4.0, A2DP, EDR
+bluetooth: {spec: '4.0', profiles: [A2DP]}
 cameras:
 - {flash: 'LED', info: '13 MP'}
 - {flash: '', info: '5 MP'}
diff --git a/_data/devices/Z00D.yml b/_data/devices/Z00D.yml
index 0168170..2c0de38 100644
--- a/_data/devices/Z00D.yml
+++ b/_data/devices/Z00D.yml
@@ -1,6 +1,6 @@
 architecture: x86
 battery: 2500mAh
-bluetooth: v4.0 with A2DP, EDR
+bluetooth: {spec: '4.0', profiles: [A2DP]}
 cameras:
 - {flash: 'LED', info: '8 MP'}
 - {flash: '', info: '2 MP'}
diff --git a/_data/devices/Z00L.yml b/_data/devices/Z00L.yml
index 92712fd..389a650 100644
--- a/_data/devices/Z00L.yml
+++ b/_data/devices/Z00L.yml
@@ -1,6 +1,6 @@
 architecture: arm64
 battery: 3000 mAh
-bluetooth: v4.1 with A2DP, LE
+bluetooth: {spec: '4.1', profiles: [A2DP]}
 cameras:
 - {flash: 'LED', info: '13 MP'}
 - {flash: '', info: '5 MP'}
diff --git a/_data/devices/Z00T.yml b/_data/devices/Z00T.yml
index d38d88c..8dec11d 100644
--- a/_data/devices/Z00T.yml
+++ b/_data/devices/Z00T.yml
@@ -1,6 +1,6 @@
 architecture: arm64
 battery: 3000 mAh
-bluetooth: v4.1 with A2DP, LE
+bluetooth: {spec: '4.1', profiles: [A2DP]}
 cameras:
 - {flash: 'LED', info: '13 MP'}
 - {flash: '', info: '5 MP, 13 MP Flash LED (ZD551KL)'}
diff --git a/_data/devices/addison.yml b/_data/devices/addison.yml
index 56873b7..01120ca 100644
--- a/_data/devices/addison.yml
+++ b/_data/devices/addison.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Non-removable 3510 mAh
-bluetooth: v4.0 with A2DP, LE
+bluetooth: {spec: 4.0 + LE, profiles: [A2DP]}
 cameras:
 - {flash: 'Dual LED', info: '16 MP'}
 - {flash: 'LED', info: '5 MP'}
diff --git a/_data/devices/angler.yml b/_data/devices/angler.yml
index a8ef08e..a5f0f73 100644
--- a/_data/devices/angler.yml
+++ b/_data/devices/angler.yml
@@ -1,6 +1,6 @@
 architecture: arm64
 battery: 3450 mAh, Li-Po non-removable
-bluetooth: 4.2 with A2DP, LE
+bluetooth: {spec: '4.2', profiles: [A2DP]}
 cameras:
 - {flash: 'Dual LED (dual tone)', info: '12.3 MP, laser autofocus'}
 - {flash: '', info: '8 MP'}
diff --git a/_data/devices/armani.yml b/_data/devices/armani.yml
index f235174..befb6fa 100644
--- a/_data/devices/armani.yml
+++ b/_data/devices/armani.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Removable Li-Ion 2000 mAh
-bluetooth: v4.0 with A2DP, LE
+bluetooth: {spec: 4.0 + LE, profiles: [A2DP]}
 cameras:
 - {flash: 'LED', info: '8 MP'}
 - {flash: '', info: '1.6 MP'}
diff --git a/_data/devices/athene.yml b/_data/devices/athene.yml
index 0a2b21d..d3da028 100644
--- a/_data/devices/athene.yml
+++ b/_data/devices/athene.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: 3000 mAh
-bluetooth: 4.1 LE
+bluetooth: {spec: '4.1'}
 cameras:
 - {flash: 'Dual LED', info: '16 MP'}
 - {flash: '', info: '5 MP'}
diff --git a/_data/devices/axon7.yml b/_data/devices/axon7.yml
index f76d98d..be29c8e 100644
--- a/_data/devices/axon7.yml
+++ b/_data/devices/axon7.yml
@@ -1,6 +1,6 @@
 architecture: arm64
 battery: Non-removable Li-Ion 3250 mAh
-bluetooth: v4.2 with A2DP, LE
+bluetooth: {spec: '4.2', profiles: [A2DP]}
 cameras:
 - {flash: 'Dual LED (dual tone)', info: '20 MP'}
 - {flash: '', info: '8 MP'}
diff --git a/_data/devices/bacon.yml b/_data/devices/bacon.yml
index d5c2265..0e43e57 100644
--- a/_data/devices/bacon.yml
+++ b/_data/devices/bacon.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: 3100 mAh
-bluetooth: '4.1'
+bluetooth: {spec: '4.1'}
 cameras:
 - {flash: 'Dual LED', info: '13 MP  F/2.0'}
 - {flash: '', info: '5 MP'}
diff --git a/_data/devices/bullhead.yml b/_data/devices/bullhead.yml
index 0234b4e..9c576c3 100644
--- a/_data/devices/bullhead.yml
+++ b/_data/devices/bullhead.yml
@@ -1,6 +1,6 @@
 architecture: arm64
 battery: Non-removable Li-Po 2700 mAh
-bluetooth: v4.2 with A2DP, LE
+bluetooth: {spec: '4.2', profiles: [A2DP]}
 cameras:
 - {flash: 'Dual LED (dual tone)', info: '12.3 MP, laser autofocus'}
 - {flash: '', info: '5 MP'}
diff --git a/_data/devices/cancro.yml b/_data/devices/cancro.yml
index 0b1fef6..6a48f95 100644
--- a/_data/devices/cancro.yml
+++ b/_data/devices/cancro.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Non-removable Li-Ion  3050 mAh (Mi 3) / 3080 mAh (Mi 4)
-bluetooth: v4.0 with A2DP
+bluetooth: {spec: '4.0', profiles: [A2DP]}
 cameras:
 - {flash: 'LED', info: '13 MP Sony IMX135 Exmor RS (Mi 3) / 13 MP Sony IMX214 Exmor RS (Mi 4)'}
 - {flash: '', info: '2.07 MP  (Mi 3) / 7.99 MP (Mi 4)'}
diff --git a/_data/devices/capricorn.yml b/_data/devices/capricorn.yml
index b14700a..080744a 100644
--- a/_data/devices/capricorn.yml
+++ b/_data/devices/capricorn.yml
@@ -1,6 +1,6 @@
 architecture: arm64
 battery: Non-removable Li-Po 3200 mAh
-bluetooth: v4.2 with A2DP, LE, aptX
+bluetooth: {spec: '4.2', profiles: [A2DP + aptX]}
 cameras:
 - {flash: 'Dual LED (dual tone)', info: '12 MP'}
 - {flash: '', info: '4 MP'}
diff --git a/_data/devices/chagallwifi.yml b/_data/devices/chagallwifi.yml
index 2e353ed..d25e1f2 100644
--- a/_data/devices/chagallwifi.yml
+++ b/_data/devices/chagallwifi.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Non-removable Li-Ion 7900 mAh
-bluetooth: v4.0 with A2DP
+bluetooth: {spec: '4.0', profiles: [A2DP]}
 cameras:
 - {flash: 'LED', info: '8 MP'}
 - {flash: '', info: '2.1 MP'}
diff --git a/_data/devices/che10.yml b/_data/devices/che10.yml
index 0e9a986..e5b51c6 100644
--- a/_data/devices/che10.yml
+++ b/_data/devices/che10.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Li-Ion 2000 / 3000 mAh
-bluetooth: '4.0'
+bluetooth: {spec: '4.0'}
 cameras:
 - {flash: 'LED', info: '8 / 13 MP'}
 - {flash: '', info: '2 / 5 MP'}
diff --git a/_data/devices/cheeseburger.yml b/_data/devices/cheeseburger.yml
index 58056cf..ff66bad 100644
--- a/_data/devices/cheeseburger.yml
+++ b/_data/devices/cheeseburger.yml
@@ -1,6 +1,6 @@
 architecture: arm64
 battery: Non-removable Li-Ion 3300 mAh
-bluetooth: v4.2 with A2DP, LE
+bluetooth: {spec: '4.2', profiles: [A2DP]}
 cameras:
 - {flash: 'LED', info: '20 MP'}
 - {flash: 'LED', info: '16 MP'}
diff --git a/_data/devices/cherry.yml b/_data/devices/cherry.yml
index 6917260..047b985 100644
--- a/_data/devices/cherry.yml
+++ b/_data/devices/cherry.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Li-Ion 2000 / 3000 mAh
-bluetooth: '4.0'
+bluetooth: {spec: '4.0'}
 cameras:
 - {flash: 'LED', info: '8 / 13 MP'}
 - {flash: '', info: '2 / 5 MP'}
diff --git a/_data/devices/clark.yml b/_data/devices/clark.yml
index 497fb1d..731bbab 100644
--- a/_data/devices/clark.yml
+++ b/_data/devices/clark.yml
@@ -1,6 +1,6 @@
 architecture: arm64
 battery: 3000 mAh
-bluetooth: 4.1 LE
+bluetooth: {spec: '4.1'}
 cameras:
 - {flash: 'Dual LED', info: '21 MP'}
 - {flash: '', info: '5 MP'}
diff --git a/_data/devices/condor.yml b/_data/devices/condor.yml
index da9aee5..c8192dd 100644
--- a/_data/devices/condor.yml
+++ b/_data/devices/condor.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: 1980 mAh
-bluetooth: v4.0
+bluetooth: {spec: '4.0'}
 cameras:
 - {flash: '', info: '5 MP'}
 carrier:
diff --git a/_data/devices/crackling.yml b/_data/devices/crackling.yml
index 8a7f62f..96f10c0 100644
--- a/_data/devices/crackling.yml
+++ b/_data/devices/crackling.yml
@@ -1,6 +1,6 @@
 architecture: arm64
 battery: Removable 2500 mAh
-bluetooth: v4, A2DP, BLE
+bluetooth: {spec: 4.0 + LE, profiles: [A2DP]}
 cameras:
 - {flash: 'Dual LED', info: '13 MP'}
 - {flash: '', info: '5 MP'}
diff --git a/_data/devices/d2att.yml b/_data/devices/d2att.yml
index c00c4da..3383983 100644
--- a/_data/devices/d2att.yml
+++ b/_data/devices/d2att.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: 2100 mAh
-bluetooth: '4.0'
+bluetooth: {spec: '4.0'}
 cameras:
 - {flash: '', info: '8 MP w/ LED flash'}
 - {flash: '', info: '1.9 MP'}
diff --git a/_data/devices/d2spr.yml b/_data/devices/d2spr.yml
index 8a0970b..7cc27a6 100644
--- a/_data/devices/d2spr.yml
+++ b/_data/devices/d2spr.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: 2100 mAh
-bluetooth: '4.0'
+bluetooth: {spec: '4.0'}
 cameras:
 - {flash: '', info: '8 MP w/ LED flash'}
 - {flash: '', info: '1.9 MP'}
diff --git a/_data/devices/d2tmo.yml b/_data/devices/d2tmo.yml
index ec70e18..1f42a6e 100644
--- a/_data/devices/d2tmo.yml
+++ b/_data/devices/d2tmo.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: 2100 mAh
-bluetooth: '4.0'
+bluetooth: {spec: '4.0'}
 cameras:
 - {flash: '', info: '8 MP w/ LED flash'}
 - {flash: '', info: '1.9 MP'}
diff --git a/_data/devices/d2vzw.yml b/_data/devices/d2vzw.yml
index 1b8f899..fa85ba5 100644
--- a/_data/devices/d2vzw.yml
+++ b/_data/devices/d2vzw.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: 2100 mAh
-bluetooth: '4.0'
+bluetooth: {spec: '4.0'}
 cameras:
 - {flash: '', info: '8 MP w/ LED flash'}
 - {flash: '', info: '1.9 MP'}
diff --git a/_data/devices/d800.yml b/_data/devices/d800.yml
index f5c3a99..b79573e 100644
--- a/_data/devices/d800.yml
+++ b/_data/devices/d800.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Non-removable Li-Po 3000 mAh
-bluetooth: v4.0 with A2DP, LE, aptX
+bluetooth: {spec: 4.0 + LE, profiles: [A2DP + aptX]}
 cameras:
 - {flash: 'LED', info: '13 MP'}
 - {flash: '', info: '2.1 MP'}
diff --git a/_data/devices/d801.yml b/_data/devices/d801.yml
index d043625..7e56682 100644
--- a/_data/devices/d801.yml
+++ b/_data/devices/d801.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Non-removable Li-Po 3000 mAh
-bluetooth: v4.0 with A2DP, LE, aptX
+bluetooth: {spec: 4.0 + LE, profiles: [A2DP + aptX]}
 cameras:
 - {flash: 'LED', info: '13 MP'}
 - {flash: '', info: '2.1 MP'}
diff --git a/_data/devices/d802.yml b/_data/devices/d802.yml
index b170c16..1ab00c8 100644
--- a/_data/devices/d802.yml
+++ b/_data/devices/d802.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Non-removable Li-Po 3000 mAh
-bluetooth: v4.0 with A2DP, LE, aptX
+bluetooth: {spec: 4.0 + LE, profiles: [A2DP + aptX]}
 cameras:
 - {flash: 'LED', info: '13 MP'}
 - {flash: '', info: '2.1 MP'}
diff --git a/_data/devices/d803.yml b/_data/devices/d803.yml
index a442e41..dc4b17d 100644
--- a/_data/devices/d803.yml
+++ b/_data/devices/d803.yml
@@ -1,6 +1,6 @@
 architecture: arm64
 battery: Non-removable Li-Po 3000 mAh
-bluetooth: v4.0 with A2DP, LE, aptX
+bluetooth: {spec: 4.0 + LE, profiles: [A2DP + aptX]}
 cameras:
 - {flash: 'LED', info: '13 MP'}
 - {flash: '', info: '2.1 MP'}
diff --git a/_data/devices/d850.yml b/_data/devices/d850.yml
index e8de838..2cd1326 100644
--- a/_data/devices/d850.yml
+++ b/_data/devices/d850.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Removable Li-Ion 3000 mAh
-bluetooth: v4.0 with A2DP, LE, aptX
+bluetooth: {spec: 4.0 + LE, profiles: [A2DP + aptX]}
 cameras:
 - {flash: 'Dual LED', info: '13 MP'}
 - {flash: '', info: '2.1 MP'}
diff --git a/_data/devices/d851.yml b/_data/devices/d851.yml
index 6a618bf..1485462 100644
--- a/_data/devices/d851.yml
+++ b/_data/devices/d851.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Removable Li-Ion 3000 mAh
-bluetooth: v4.0 with A2DP, LE, aptX
+bluetooth: {spec: 4.0 + LE, profiles: [A2DP + aptX]}
 cameras:
 - {flash: 'Dual LED', info: '13 MP'}
 - {flash: '', info: '2.1 MP'}
diff --git a/_data/devices/d852.yml b/_data/devices/d852.yml
index e365dff..26ceb14 100644
--- a/_data/devices/d852.yml
+++ b/_data/devices/d852.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Removable Li-Ion 3000 mAh
-bluetooth: v4.0 with A2DP, LE, aptX
+bluetooth: {spec: 4.0 + LE, profiles: [A2DP + aptX]}
 cameras:
 - {flash: 'Dual LED', info: '13 MP'}
 - {flash: '', info: '2.1 MP'}
diff --git a/_data/devices/d855.yml b/_data/devices/d855.yml
index 94ea1a5..6a09c3d 100644
--- a/_data/devices/d855.yml
+++ b/_data/devices/d855.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Removable Li-Ion 3000 mAh
-bluetooth: v4.0 with A2DP, LE, aptX
+bluetooth: {spec: 4.0 + LE, profiles: [A2DP + aptX]}
 cameras:
 - {flash: 'Dual LED', info: '13 MP'}
 - {flash: '', info: '2.1 MP'}
diff --git a/_data/devices/deb.yml b/_data/devices/deb.yml
index 32e268a..366cfda 100644
--- a/_data/devices/deb.yml
+++ b/_data/devices/deb.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: 3950 mAh
-bluetooth: 4.0 LE
+bluetooth: {spec: 4.0 + LE}
 cameras:
 - {flash: '', info: '5 MP'}
 - {flash: '', info: '1.2 MP'}
diff --git a/_data/devices/dogo.yml b/_data/devices/dogo.yml
index dd2a7f8..9b6a034 100644
--- a/_data/devices/dogo.yml
+++ b/_data/devices/dogo.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Removable 2300 mAh
-bluetooth: v4.0 with A2DP, LE
+bluetooth: {spec: 4.0 + LE, profiles: [A2DP]}
 cameras:
 - {flash: 'LED', info: '13.1MP f/2.4'}
 - {flash: '', info: 'VGA'}
diff --git a/_data/devices/dragon.yml b/_data/devices/dragon.yml
index 232166a..fb0e064 100644
--- a/_data/devices/dragon.yml
+++ b/_data/devices/dragon.yml
@@ -1,6 +1,6 @@
 architecture: arm64
 battery: Non-removable Li-Po 9240 mAh
-bluetooth: v4.1 with A2DP
+bluetooth: {spec: '4.1', profiles: [A2DP]}
 cameras:
 - {flash: '', info: '8 MP, f/2.4'}
 - {flash: '', info: '2 MP'}
diff --git a/_data/devices/espresso3g.yml b/_data/devices/espresso3g.yml
index d0e01ea..ee70f3b 100644
--- a/_data/devices/espresso3g.yml
+++ b/_data/devices/espresso3g.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Non-removable Li-Ion 4000 mAh / Li-Po 7000 mAh
-bluetooth: v3.0 with A2DP
+bluetooth: {spec: '3.0', profiles: [A2DP]}
 cameras:
 - {flash: '', info: '15 MP '}
 - {flash: '', info: '1.3 MP'}
diff --git a/_data/devices/espressowifi.yml b/_data/devices/espressowifi.yml
index d06effd..dd18d5d 100644
--- a/_data/devices/espressowifi.yml
+++ b/_data/devices/espressowifi.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Non-removable Li-Ion 4000 mAh / Li-Po 7000 mAh
-bluetooth: v3.0 with A2DP
+bluetooth: {spec: '3.0', profiles: [A2DP]}
 cameras:
 - {flash: '', info: '15 MP '}
 - {flash: '', info: '1.3 MP'}
diff --git a/_data/devices/ether.yml b/_data/devices/ether.yml
index 31fa63c..10e7f96 100644
--- a/_data/devices/ether.yml
+++ b/_data/devices/ether.yml
@@ -1,6 +1,6 @@
 architecture: arm64
 battery: Non-removable Li-Ion 2680 mAh
-bluetooth: 4.0 LE
+bluetooth: {spec: 4.0 + LE}
 cameras:
 - {flash: 'LED (dual-tone)', info: '13 MP'}
 - {flash: '', info: '5 MP'}
diff --git a/_data/devices/f400.yml b/_data/devices/f400.yml
index 26b7021..22727dc 100644
--- a/_data/devices/f400.yml
+++ b/_data/devices/f400.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Removable Li-Ion 3000 mAh
-bluetooth: v4.0 with A2DP, LE, aptX
+bluetooth: {spec: 4.0 + LE, profiles: [A2DP + aptX]}
 cameras:
 - {flash: 'Dual LED', info: '13 MP'}
 - {flash: '', info: '2.1 MP'}
diff --git a/_data/devices/falcon.yml b/_data/devices/falcon.yml
index 1ecead6..ddd0d88 100644
--- a/_data/devices/falcon.yml
+++ b/_data/devices/falcon.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: 2070 mAh
-bluetooth: v4.0 with A2DP, LE
+bluetooth: {spec: 4.0 + LE, profiles: [A2DP]}
 cameras:
 - {flash: 'LED', info: '5 MP'}
 - {flash: '', info: '1.3 MP'}
diff --git a/_data/devices/find7.yml b/_data/devices/find7.yml
index d5cc586..2910704 100644
--- a/_data/devices/find7.yml
+++ b/_data/devices/find7.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Removable 2800 mAh Li-Po with Rapid Charge
-bluetooth: '4.0'
+bluetooth: {spec: '4.0'}
 cameras:
 - {flash: 'Dual LED', info: '13 MP'}
 - {flash: '', info: '5 MP'}
diff --git a/_data/devices/find7s.yml b/_data/devices/find7s.yml
index 75b03a4..d29a539 100644
--- a/_data/devices/find7s.yml
+++ b/_data/devices/find7s.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Removable 3000 mAh Li-Po with Rapid Charge
-bluetooth: '4.0'
+bluetooth: {spec: '4.0'}
 cameras:
 - {flash: 'Dual LED', info: '13 MP'}
 - {flash: '', info: '5 MP'}
diff --git a/_data/devices/flo.yml b/_data/devices/flo.yml
index 0c5a797..2ccf14a 100644
--- a/_data/devices/flo.yml
+++ b/_data/devices/flo.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: 3950 mAh
-bluetooth: 4.0 LE
+bluetooth: {spec: 4.0 + LE}
 cameras:
 - {flash: '', info: '5 MP'}
 - {flash: '', info: '1.2 MP'}
diff --git a/_data/devices/flounder.yml b/_data/devices/flounder.yml
index ece2f46..2f9819a 100644
--- a/_data/devices/flounder.yml
+++ b/_data/devices/flounder.yml
@@ -1,6 +1,6 @@
 architecture: arm64
 battery: Non-removable Li-Po 6700 mAh
-bluetooth: v4.1 with A2DP
+bluetooth: {spec: '4.1', profiles: [A2DP]}
 cameras:
 - {flash: 'LED', info: '8 MP'}
 - {flash: '', info: '1.6 MP'}
diff --git a/_data/devices/flounder_lte.yml b/_data/devices/flounder_lte.yml
index 4dcaf85..64427a1 100644
--- a/_data/devices/flounder_lte.yml
+++ b/_data/devices/flounder_lte.yml
@@ -1,6 +1,6 @@
 architecture: arm64
 battery: Non-removable Li-Po 6700 mAh
-bluetooth: v4.1 with A2DP
+bluetooth: {spec: '4.1', profiles: [A2DP]}
 cameras:
 - {flash: 'LED', info: '8 MP'}
 - {flash: '', info: '1.6 MP'}
diff --git a/_data/devices/foster.yml b/_data/devices/foster.yml
index 788bed9..0d459e8 100644
--- a/_data/devices/foster.yml
+++ b/_data/devices/foster.yml
@@ -1,6 +1,6 @@
 architecture: arm64
 battery:
-bluetooth: v4.1, LE
+bluetooth: {spec: '4.1'}
 cameras: []
 carrier:
 channels: [weekly]
diff --git a/_data/devices/fugu.yml b/_data/devices/fugu.yml
index d4839ef..b4d6c71 100644
--- a/_data/devices/fugu.yml
+++ b/_data/devices/fugu.yml
@@ -1,6 +1,6 @@
 architecture: x86_64
 battery:
-bluetooth: v4.1, LE
+bluetooth: {spec: '4.1'}
 cameras: []
 carrier:
 channels: [weekly]
diff --git a/_data/devices/g2m.yml b/_data/devices/g2m.yml
index 87744a8..398212f 100644
--- a/_data/devices/g2m.yml
+++ b/_data/devices/g2m.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Removable Li-Ion 2440 mAh
-bluetooth: v4.0
+bluetooth: {spec: '4.0'}
 cameras:
 - {flash: 'LED', info: '8 MP'}
 - {flash: '', info: '1.3 MP'}
diff --git a/_data/devices/gemini.yml b/_data/devices/gemini.yml
index 5832b3e..14153e9 100644
--- a/_data/devices/gemini.yml
+++ b/_data/devices/gemini.yml
@@ -1,6 +1,6 @@
 architecture: arm64
 battery: Non-removable Li-Po 3000 mAh
-bluetooth: v4.2 with A2DP, LE, aptX
+bluetooth: {spec: '4.2', profiles: [A2DP + aptX]}
 cameras:
 - {flash: 'Dual LED (dual tone)', info: '16 MP'}
 - {flash: '', info: '4 MP'}
diff --git a/_data/devices/ghost.yml b/_data/devices/ghost.yml
index 4a9519e..63db2e0 100644
--- a/_data/devices/ghost.yml
+++ b/_data/devices/ghost.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Non-removable Li-Ion 2200 mAh
-bluetooth: v4.0 with A2DP, EDR, LE
+bluetooth: {spec: 4.0 + LE, profiles: [A2DP]}
 cameras:
 - {flash: 'LED', info: '10 MP'}
 - {flash: '', info: '2 MP'}
diff --git a/_data/devices/gohan.yml b/_data/devices/gohan.yml
index 7aed46e..956e923 100644
--- a/_data/devices/gohan.yml
+++ b/_data/devices/gohan.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Non-removable Li-Po 3100 mAh
-bluetooth: v4.0
+bluetooth: {spec: '4.0'}
 cameras:
 - {flash: 'Dual LED', info: '16 MP'}
 - {flash: '', info: '8 MP'}
diff --git a/_data/devices/griffin.yml b/_data/devices/griffin.yml
index 820d9db..976b73d 100644
--- a/_data/devices/griffin.yml
+++ b/_data/devices/griffin.yml
@@ -1,6 +1,6 @@
 architecture: arm64
 battery: Non-removable Li-Ion 2600 mAh
-bluetooth: v4.1 with A2DP, LE
+bluetooth: {spec: '4.1', profiles: [A2DP]}
 cameras:
 - {flash: 'Dual LED (dual tone)', info: '13 MP, laser autofocus, OIS'}
 - {flash: 'LED', info: '5 MP'}
diff --git a/_data/devices/gtelwifiue.yml b/_data/devices/gtelwifiue.yml
index 62dc03b..7699c6c 100644
--- a/_data/devices/gtelwifiue.yml
+++ b/_data/devices/gtelwifiue.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Removable Li-Ion 7300 mAh
-bluetooth: v4.1 with A2DP
+bluetooth: {spec: '4.1', profiles: [A2DP]}
 cameras:
 - {flash: '', info: '5 MP'}
 - {flash: '', info: '2 MP'}
diff --git a/_data/devices/gtesqltespr.yml b/_data/devices/gtesqltespr.yml
index f750f17..c0ae2a5 100644
--- a/_data/devices/gtesqltespr.yml
+++ b/_data/devices/gtesqltespr.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Removable Li-Ion 5000 mAh
-bluetooth: v4.1 with A2DP
+bluetooth: {spec: '4.1', profiles: [A2DP]}
 cameras:
 - {flash: '', info: '5 MP'}
 - {flash: '', info: '2 MP'}
diff --git a/_data/devices/gts210ltexx.yml b/_data/devices/gts210ltexx.yml
index 2067b99..1396f5a 100644
--- a/_data/devices/gts210ltexx.yml
+++ b/_data/devices/gts210ltexx.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Non-removable Li-Ion 5870 mAh
-bluetooth: v4.1 with A2DP, LE
+bluetooth: {spec: '4.1', profiles: [A2DP]}
 cameras:
 - {flash: '', info: '8 MP'}
 - {flash: '', info: '2.1 MP'}
diff --git a/_data/devices/gts210vewifi.yml b/_data/devices/gts210vewifi.yml
index 944b10c..3bec27b 100644
--- a/_data/devices/gts210vewifi.yml
+++ b/_data/devices/gts210vewifi.yml
@@ -1,6 +1,6 @@
 architecture: arm64
 battery: Non-removable Li-Ion 5870 mAh
-bluetooth: v4.1 with A2DP, LE
+bluetooth: {spec: '4.1', profiles: [A2DP]}
 cameras:
 - {flash: '', info: '8 MP'}
 - {flash: '', info: '2.1 MP'}
diff --git a/_data/devices/gts210wifi.yml b/_data/devices/gts210wifi.yml
index 6b7d9b3..0823a4b 100644
--- a/_data/devices/gts210wifi.yml
+++ b/_data/devices/gts210wifi.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Non-removable Li-Ion 5870 mAh
-bluetooth: v4.1 with A2DP, LE
+bluetooth: {spec: '4.1', profiles: [A2DP]}
 cameras:
 - {flash: '', info: '8 MP'}
 - {flash: '', info: '2.1 MP'}
diff --git a/_data/devices/gts28vewifi.yml b/_data/devices/gts28vewifi.yml
index a6ef57a..827d0f5 100644
--- a/_data/devices/gts28vewifi.yml
+++ b/_data/devices/gts28vewifi.yml
@@ -1,6 +1,6 @@
 architecture: arm64
 battery: Non-removable Li-Ion 4000 mAh
-bluetooth: v4.1 with A2DP, LE
+bluetooth: {spec: '4.1', profiles: [A2DP]}
 cameras:
 - {flash: '', info: '8 MP'}
 - {flash: '', info: '2.1 MP'}
diff --git a/_data/devices/h811.yml b/_data/devices/h811.yml
index 40da08d..4a00b62 100644
--- a/_data/devices/h811.yml
+++ b/_data/devices/h811.yml
@@ -1,6 +1,6 @@
 architecture: arm64
 battery: Removable Li-Ion 3000 mAh
-bluetooth: v4.1 with A2DP, LE, aptX
+bluetooth: {spec: '4.1', profiles: [A2DP + aptX]}
 cameras:
 - {flash: 'LED', info: '16 MP'}
 - {flash: '', info: '8 MP'}
diff --git a/_data/devices/h815.yml b/_data/devices/h815.yml
index c396d5b..5e9aafc 100644
--- a/_data/devices/h815.yml
+++ b/_data/devices/h815.yml
@@ -1,6 +1,6 @@
 architecture: arm64
 battery: Removable Li-Ion 3000 mAh
-bluetooth: v4.1 with A2DP, LE, aptX
+bluetooth: {spec: '4.1', profiles: [A2DP + aptX]}
 cameras:
 - {flash: 'LED', info: '16 MP'}
 - {flash: '', info: '8 MP'}
diff --git a/_data/devices/h830.yml b/_data/devices/h830.yml
index 77c20dc..30f5d38 100644
--- a/_data/devices/h830.yml
+++ b/_data/devices/h830.yml
@@ -1,6 +1,6 @@
 architecture: arm64
 battery: Removable Li-Ion 2800 mAh
-bluetooth: v4.2 with A2DP, LE, aptX HD
+bluetooth: {spec: '4.2', profiles: [A2DP + aptX HD]}
 cameras:
 - {flash: 'LED', info: '16 MP'}
 - {flash: 'LED', info: '8 MP'}
diff --git a/_data/devices/h850.yml b/_data/devices/h850.yml
index 6551c64..a00c689 100644
--- a/_data/devices/h850.yml
+++ b/_data/devices/h850.yml
@@ -1,6 +1,6 @@
 architecture: arm64
 battery: Removable Li-Ion 2800 mAh
-bluetooth: v4.2 with A2DP, LE, aptX HD
+bluetooth: {spec: '4.2', profiles: [A2DP + aptX HD]}
 cameras:
 - {flash: 'LED', info: '16 MP'}
 - {flash: 'LED', info: '8 MP'}
diff --git a/_data/devices/h870.yml b/_data/devices/h870.yml
index 00237e2..57c121f 100644
--- a/_data/devices/h870.yml
+++ b/_data/devices/h870.yml
@@ -1,6 +1,6 @@
 architecture: arm64
 battery: Removable Li-Ion 3300 mAh
-bluetooth: v4.2 with A2DP, LE, aptX HD
+bluetooth: {spec: '4.2', profiles: [A2DP + aptX HD]}
 cameras:
 - {flash: 'LED', info: '13 MP'}
 - {flash: 'LED', info: '13 MP'}
diff --git a/_data/devices/h910.yml b/_data/devices/h910.yml
index 068cbd3..437f71a 100644
--- a/_data/devices/h910.yml
+++ b/_data/devices/h910.yml
@@ -1,6 +1,6 @@
 architecture: arm64
 battery: Removable Li-Ion 3200 mAh
-bluetooth: v4.2 with A2DP, LE, aptX HD
+bluetooth: {spec: '4.2', profiles: [A2DP + aptX HD]}
 cameras:
 - {flash: 'LED', info: '16 MP'}
 - {flash: 'LED', info: '8 MP'}
diff --git a/_data/devices/h918.yml b/_data/devices/h918.yml
index 62fe1bb..8137f6f 100644
--- a/_data/devices/h918.yml
+++ b/_data/devices/h918.yml
@@ -1,6 +1,6 @@
 architecture: arm64
 battery: Removable Li-Ion 3200 mAh
-bluetooth: v4.2 with A2DP, LE, aptX HD
+bluetooth: {spec: '4.2', profiles: [A2DP + aptX HD]}
 cameras:
 - {flash: 'LED', info: '16 MP'}
 - {flash: 'LED', info: '8 MP'}
diff --git a/_data/devices/ha3g.yml b/_data/devices/ha3g.yml
index 4f89ef2..27c54af 100644
--- a/_data/devices/ha3g.yml
+++ b/_data/devices/ha3g.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Removable Li-Ion 3200 mAh
-bluetooth: v4.0 with A2DP, LE, EDR
+bluetooth: {spec: 4.0 + LE, profiles: [A2DP]}
 cameras:
 - {flash: 'LED', info: '13 MP, 1080p Full HD Video Recording @ 60fps'}
 - {flash: '', info: '2 MP, 1080p HD Video Recording @ 30fps'}
diff --git a/_data/devices/ham.yml b/_data/devices/ham.yml
index 239252b..c5d5f3c 100644
--- a/_data/devices/ham.yml
+++ b/_data/devices/ham.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Non-removable Li-Po 4100 mAh
-bluetooth: v4.1
+bluetooth: {spec: '4.1'}
 cameras:
 - {flash: 'Dual LED', info: '13 MP F/2.2'}
 - {flash: '', info: '8 MP'}
diff --git a/_data/devices/hammerhead.yml b/_data/devices/hammerhead.yml
index fb984f1..841d56f 100644
--- a/_data/devices/hammerhead.yml
+++ b/_data/devices/hammerhead.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: 2300 mAh
-bluetooth: 4.0 LE
+bluetooth: {spec: 4.0 + LE}
 cameras:
 - {flash: 'LED', info: '8 MP'}
 - {flash: '', info: '1.3 MP'}
diff --git a/_data/devices/harpia.yml b/_data/devices/harpia.yml
index 356b45e..4590039 100644
--- a/_data/devices/harpia.yml
+++ b/_data/devices/harpia.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: 2800 mAh
-bluetooth: v4.1 with A2DP, LE
+bluetooth: {spec: '4.1', profiles: [A2DP]}
 cameras:
 - {flash: 'LED', info: '8 MP'}
 - {flash: '', info: '5 MP'}
diff --git a/_data/devices/hayabusa.yml b/_data/devices/hayabusa.yml
index 07287a5..ace5e4d 100644
--- a/_data/devices/hayabusa.yml
+++ b/_data/devices/hayabusa.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Removable 1750 mAh
-bluetooth: v4.0 with A2DP
+bluetooth: {spec: '4.0', profiles: [A2DP]}
 cameras:
 - {flash: 'LED', info: '13 MP f/2.4'}
 - {flash: '', info: '1.3 MP'}
diff --git a/_data/devices/hero2lte.yml b/_data/devices/hero2lte.yml
index 83468b6..e038f37 100644
--- a/_data/devices/hero2lte.yml
+++ b/_data/devices/hero2lte.yml
@@ -1,6 +1,6 @@
 architecture: arm64
 battery: Non-removable Li-Ion 3600 mAh
-bluetooth: 4.2 with A2DP, LE, apt-X
+bluetooth: {spec: '4.2', profiles: [A2DP + aptX]}
 cameras:
 - {flash: 'LED', info: '12 MP , f/1.7, 26mm'}
 - {flash: '', info: '5 MP, f/1.7, 22mm'}
diff --git a/_data/devices/herolte.yml b/_data/devices/herolte.yml
index 328b287..73c0cdf 100644
--- a/_data/devices/herolte.yml
+++ b/_data/devices/herolte.yml
@@ -1,6 +1,6 @@
 architecture: arm64
 battery: Non-removable Li-Ion 3000 mAh
-bluetooth: 4.2 with A2DP, LE, apt-X
+bluetooth: {spec: '4.2', profiles: [A2DP + aptX]}
 cameras:
 - {flash: 'LED', info: '12 MP , f/1.7, 26mm'}
 - {flash: '', info: '5 MP, f/1.7, 22mm'}
diff --git a/_data/devices/hiaeuhl.yml b/_data/devices/hiaeuhl.yml
index 689d4a4..4e39a68 100644
--- a/_data/devices/hiaeuhl.yml
+++ b/_data/devices/hiaeuhl.yml
@@ -1,6 +1,6 @@
 architecture: arm64
 battery: Non-removable Li-Ion 2150 mAh
-bluetooth: v4.1 with A2DP
+bluetooth: {spec: '4.1', profiles: [A2DP]}
 cameras:
 - {flash: 'Dual-LED', info: '13 MP'}
 - {flash: '', info: '4 MP'}
diff --git a/_data/devices/hiaeul.yml b/_data/devices/hiaeul.yml
index 1cf38b8..6545ecd 100644
--- a/_data/devices/hiaeul.yml
+++ b/_data/devices/hiaeul.yml
@@ -1,6 +1,6 @@
 architecture: arm64
 battery: Non-removable Li-Ion 2150 mAh
-bluetooth: v4.1 with A2DP
+bluetooth: {spec: '4.1', profiles: [A2DP]}
 cameras:
 - {flash: 'Dual-LED', info: '13 MP'}
 - {flash: '', info: '4 MP'}
diff --git a/_data/devices/himaul.yml b/_data/devices/himaul.yml
index 22d286b..2a24b33 100644
--- a/_data/devices/himaul.yml
+++ b/_data/devices/himaul.yml
@@ -1,6 +1,6 @@
 architecture: arm64
 battery: 2840 mAh (non-removable)
-bluetooth: v4.1 with A2DP, aptX
+bluetooth: {spec: '4.1', profiles: [A2DP + aptX]}
 cameras:
 - {flash: 'Dual LED', info: '20 MP'}
 - {flash: '', info: '4 MP'}
diff --git a/_data/devices/himawl.yml b/_data/devices/himawl.yml
index 79da10f..05fbf68 100644
--- a/_data/devices/himawl.yml
+++ b/_data/devices/himawl.yml
@@ -1,6 +1,6 @@
 architecture: arm64
 battery: 2840 mAh (non-removable)
-bluetooth: v4.1 with A2DP, aptX
+bluetooth: {spec: '4.1', profiles: [A2DP + aptX]}
 cameras:
 - {flash: 'Dual LED', info: '20 MP'}
 - {flash: '', info: '4 MP'}
diff --git a/_data/devices/hlte.yml b/_data/devices/hlte.yml
index 2700ece..a141a04 100644
--- a/_data/devices/hlte.yml
+++ b/_data/devices/hlte.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Removable Li-Ion 3200 mAh
-bluetooth: v4.0 with A2DP, LE, EDR
+bluetooth: {spec: 4.0 + LE, profiles: [A2DP]}
 cameras:
 - {flash: 'LED', info: '13 MP, 2160p@30fps, 1080p@60fps'}
 - {flash: '', info: '2 MP, 1080p@30fps'}
diff --git a/_data/devices/hltetmo.yml b/_data/devices/hltetmo.yml
index 384346a..1e21365 100644
--- a/_data/devices/hltetmo.yml
+++ b/_data/devices/hltetmo.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Removable Li-Ion 3200 mAh
-bluetooth: v4.0 with A2DP, LE, EDR
+bluetooth: {spec: 4.0 + LE, profiles: [A2DP]}
 cameras:
 - {flash: 'LED', info: '13 MP, 2160p@30fps, 1080p@60fps'}
 - {flash: '', info: '2 MP, 1080p@30fps'}
diff --git a/_data/devices/huashan.yml b/_data/devices/huashan.yml
index ef938ab..84449d1 100644
--- a/_data/devices/huashan.yml
+++ b/_data/devices/huashan.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Non-removable 2370 mAh
-bluetooth: v4.0 with A2DP
+bluetooth: {spec: '4.0', profiles: [A2DP]}
 cameras:
 - {flash: 'LED', info: '8MP f/2.4'}
 - {flash: '', info: 'VGA'}
diff --git a/_data/devices/hydrogen.yml b/_data/devices/hydrogen.yml
index 4346157..07d7949 100644
--- a/_data/devices/hydrogen.yml
+++ b/_data/devices/hydrogen.yml
@@ -1,6 +1,6 @@
 architecture: arm64
 battery: Non-removable Li-Ion 4850 mAh
-bluetooth: v4.2 with A2DP, LE
+bluetooth: {spec: '4.2', profiles: [A2DP]}
 cameras:
 - {flash: 'LED (dual-tone)', info: '16 MP'}
 - {flash: '', info: '5 MP'}
diff --git a/_data/devices/i9100.yml b/_data/devices/i9100.yml
index 94cf023..c715e1b 100644
--- a/_data/devices/i9100.yml
+++ b/_data/devices/i9100.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: 1650 mAh
-bluetooth: 3.0+HS
+bluetooth: {spec: 3.0 + HS}
 cameras:
 - {flash: '', info: '8 MP w/ LED-flash'}
 - {flash: '', info: '1.9 MP'}
diff --git a/_data/devices/i9300.yml b/_data/devices/i9300.yml
index 66f5c29..24f1b2c 100644
--- a/_data/devices/i9300.yml
+++ b/_data/devices/i9300.yml
@@ -1,5 +1,5 @@
 architecture: arm
-bluetooth: 4.0
+bluetooth: {spec: '4.0'}
 cameras:
 - {flash: 'LED', info: '8 MP'}
 - {flash: '', info: '1.9 MP'}
diff --git a/_data/devices/i9305.yml b/_data/devices/i9305.yml
index 7f17d10..630ba8d 100644
--- a/_data/devices/i9305.yml
+++ b/_data/devices/i9305.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Removable Li-Ion 2100 mAh
-bluetooth: v4.0 with A2DP, EDR
+bluetooth: {spec: '4.0', profiles: [A2DP]}
 cameras:
 - {flash: 'LED', info: '8 MP'}
 - {flash: '', info: '1.9 MP'}
diff --git a/_data/devices/ido.yml b/_data/devices/ido.yml
index 8497751..7679862 100644
--- a/_data/devices/ido.yml
+++ b/_data/devices/ido.yml
@@ -1,6 +1,6 @@
 architecture: arm64
 battery: Non-removable Li-Ion 4100 mAh
-bluetooth: v4.1 with A2DP, LE
+bluetooth: {spec: '4.1', profiles: [A2DP]}
 cameras:
 - {flash: 'LED', info: '13 MP'}
 - {flash: '', info: '5 MP'}
diff --git a/_data/devices/ivy.yml b/_data/devices/ivy.yml
index 7e1426f..06fbbcf 100644
--- a/_data/devices/ivy.yml
+++ b/_data/devices/ivy.yml
@@ -1,6 +1,6 @@
 architecture: arm64
 battery: Non-removable Li-Ion 2930 mAh
-bluetooth: v4.1 with A2DP, aptX
+bluetooth: {spec: '4.1', profiles: [A2DP + aptX]}
 cameras:
 - {flash: 'LED', info: '20.7 MP F/2.0'}
 - {flash: '', info: '5.1 MP F/2.4'}
diff --git a/_data/devices/jag3gds.yml b/_data/devices/jag3gds.yml
index e1b54d0..91d69d8 100644
--- a/_data/devices/jag3gds.yml
+++ b/_data/devices/jag3gds.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Removable Li-Ion 2540 mAh
-bluetooth: v4.0
+bluetooth: {spec: '4.0'}
 cameras:
 - {flash: 'LED', info: '8 MP'}
 - {flash: '', info: '1.3 MP'}
diff --git a/_data/devices/jagnm.yml b/_data/devices/jagnm.yml
index 5b80e0f..2a27e05 100644
--- a/_data/devices/jagnm.yml
+++ b/_data/devices/jagnm.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Removable Li-Ion 2540 mAh
-bluetooth: v4.0
+bluetooth: {spec: '4.0'}
 cameras:
 - {flash: 'LED', info: '8 MP'}
 - {flash: '', info: '1.3 MP'}
diff --git a/_data/devices/jalebi.yml b/_data/devices/jalebi.yml
index 583e151..b9c1c30 100644
--- a/_data/devices/jalebi.yml
+++ b/_data/devices/jalebi.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: 2000 mAh
-bluetooth: '4.0'
+bluetooth: {spec: '4.0'}
 cameras:
 - {flash: '', info: '8 MP'}
 - {flash: '', info: '2 MP'}
diff --git a/_data/devices/jfltevzw.yml b/_data/devices/jfltevzw.yml
index f37afcc..b5c39af 100644
--- a/_data/devices/jfltevzw.yml
+++ b/_data/devices/jfltevzw.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Removable Li-Ion 2600 mAh
-bluetooth: 4.0
+bluetooth: {spec: '4.0'}
 cameras:
 - {flash: 'LED', info: '13 MP'}
 - {flash: '', info: '2 MP'}
diff --git a/_data/devices/jfltexx.yml b/_data/devices/jfltexx.yml
index e33f445..ec71c84 100644
--- a/_data/devices/jfltexx.yml
+++ b/_data/devices/jfltexx.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Removable Li-Ion 2600 mAh
-bluetooth: 4.0
+bluetooth: {spec: '4.0'}
 cameras:
 - {flash: 'LED', info: '13 MP'}
 - {flash: '', info: '2 MP'}
diff --git a/_data/devices/k3gxx.yml b/_data/devices/k3gxx.yml
index 4b43148..58ead1d 100644
--- a/_data/devices/k3gxx.yml
+++ b/_data/devices/k3gxx.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Removable Li-Ion 2800 mAh
-bluetooth: v4.0 with A2DP, LE, EDR
+bluetooth: {spec: 4.0 + LE, profiles: [A2DP]}
 cameras:
 - {flash: 'LED', info: '16 MP, 1080p Full HD Video Recording @ 60fps'}
 - {flash: '', info: '2 MP, 1080p HD Video Recording @ 30fps'}
diff --git a/_data/devices/karin.yml b/_data/devices/karin.yml
index c42af5b..a884419 100644
--- a/_data/devices/karin.yml
+++ b/_data/devices/karin.yml
@@ -1,6 +1,6 @@
 architecture: arm64
 battery: Non-removable Li-Ion 6000 mAh
-bluetooth: v4.1 with A2DP, aptX
+bluetooth: {spec: '4.1', profiles: [A2DP + aptX]}
 cameras:
 - {flash: 'LED', info: '8.1 MP F/2.0'}
 - {flash: '', info: '5.1 MP F/2.4'}
diff --git a/_data/devices/karin_windy.yml b/_data/devices/karin_windy.yml
index bc814b6..64d6a92 100644
--- a/_data/devices/karin_windy.yml
+++ b/_data/devices/karin_windy.yml
@@ -1,6 +1,6 @@
 architecture: arm64
 battery: Non-removable Li-Ion 6000 mAh
-bluetooth: v4.1 with A2DP, aptX
+bluetooth: {spec: '4.1', profiles: [A2DP + aptX]}
 cameras:
 - {flash: 'LED', info: '8.1 MP F/2.0'}
 - {flash: '', info: '5.1 MP F/2.4'}
diff --git a/_data/devices/kccat6.yml b/_data/devices/kccat6.yml
index 78dec94..ee7f9a3 100644
--- a/_data/devices/kccat6.yml
+++ b/_data/devices/kccat6.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: 2800 mAh Li-ion battery, removable
-bluetooth: 4.0 BLE
+bluetooth: {spec: 4.0 + LE}
 cameras:
 - {flash: 'LED', info: '16 MP'}
 - {flash: '', info: '2 MP'}
diff --git a/_data/devices/kenzo.yml b/_data/devices/kenzo.yml
index 855584e..f1c8472 100644
--- a/_data/devices/kenzo.yml
+++ b/_data/devices/kenzo.yml
@@ -1,6 +1,6 @@
 architecture: arm64
 battery: Non-removable Li-Po 4050 mAh
-bluetooth: v4.1 with A2DP, LE
+bluetooth: {spec: '4.1', profiles: [A2DP]}
 cameras:
 - {flash: 'LED (dual-tone)', info: '16 MP'}
 - {flash: '', info: '5 MP'}
diff --git a/_data/devices/kingdom.yml b/_data/devices/kingdom.yml
index e56599a..8fe6ee2 100644
--- a/_data/devices/kingdom.yml
+++ b/_data/devices/kingdom.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: 4000 mAh Li-Po, non-removable
-bluetooth: 4.0 BLE
+bluetooth: {spec: 4.0 + LE}
 cameras:
 - {flash: 'Dual LED', info: '16 MP F/2.2, OIS'}
 - {flash: '', info: '5 MP'}
diff --git a/_data/devices/kipper.yml b/_data/devices/kipper.yml
index 9baa9b5..d36cd5c 100644
--- a/_data/devices/kipper.yml
+++ b/_data/devices/kipper.yml
@@ -1,6 +1,6 @@
 architecture: arm64
 battery: 2500 mAh
-bluetooth: v4 with A2DP, BLE
+bluetooth: {spec: 4.0 + LE, profiles: [A2DP]}
 cameras:
 - {flash: 'LED', info: '20 MP'}
 - {flash: 'LED', info: '8 MP'}
diff --git a/_data/devices/kiwi.yml b/_data/devices/kiwi.yml
index d9761ce..b7ec00b 100644
--- a/_data/devices/kiwi.yml
+++ b/_data/devices/kiwi.yml
@@ -1,6 +1,6 @@
 architecture: arm64
 battery: 3000 mAh (non-removable)
-bluetooth: v4.1 with A2DP
+bluetooth: {spec: '4.1', profiles: [A2DP]}
 cameras:
 - {flash: 'LED', info: '13 MP, f/2.0, 28mm'}
 - {flash: '', info: '5 MP, f/2.4, 22mm'}
diff --git a/_data/devices/klimtwifi.yml b/_data/devices/klimtwifi.yml
index 177a6c6..b8c505d 100644
--- a/_data/devices/klimtwifi.yml
+++ b/_data/devices/klimtwifi.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Non-removable Li-Ion 4900 mAh
-bluetooth: 4.0 A2DP
+bluetooth: {spec: '4.0', profiles: [A2DP]}
 cameras:
 - {flash: 'LED', info: '8 MP'}
 - {flash: '', info: '2.1 MP'}
diff --git a/_data/devices/klte.yml b/_data/devices/klte.yml
index e2c2797..1f90249 100644
--- a/_data/devices/klte.yml
+++ b/_data/devices/klte.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: 2800 mAh Li-ion battery, removable
-bluetooth: 4.0 BLE
+bluetooth: {spec: 4.0 + LE}
 cameras:
 - {flash: 'LED', info: '16 MP'}
 - {flash: '', info: '2 MP'}
diff --git a/_data/devices/kltechn.yml b/_data/devices/kltechn.yml
index a1d1b64..6af4c65 100644
--- a/_data/devices/kltechn.yml
+++ b/_data/devices/kltechn.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: 2800 mAh Li-ion battery, removable
-bluetooth: 4.0 BLE
+bluetooth: {spec: 4.0 + LE}
 cameras:
 - {flash: 'LED', info: '16 MP'}
 - {flash: '', info: '2 MP'}
diff --git a/_data/devices/kltechnduo.yml b/_data/devices/kltechnduo.yml
index 156a103..c95665f 100644
--- a/_data/devices/kltechnduo.yml
+++ b/_data/devices/kltechnduo.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: 2800 mAh Li-ion battery, removable
-bluetooth: 4.0 BLE
+bluetooth: {spec: 4.0 + LE}
 cameras:
 - {flash: 'LED', info: '16 MP'}
 - {flash: '', info: '2 MP'}
diff --git a/_data/devices/klteduos.yml b/_data/devices/klteduos.yml
index 3fe5acb..679659b 100644
--- a/_data/devices/klteduos.yml
+++ b/_data/devices/klteduos.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: 2800 mAh Li-ion battery, removable
-bluetooth: 4.0 BLE
+bluetooth: {spec: 4.0 + LE}
 cameras:
 - {flash: 'LED', info: '16 MP'}
 - {flash: '', info: '2 MP'}
diff --git a/_data/devices/kltedv.yml b/_data/devices/kltedv.yml
index 505e7b0..c4456d1 100644
--- a/_data/devices/kltedv.yml
+++ b/_data/devices/kltedv.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: 2800 mAh Li-ion battery, removable
-bluetooth: 4.0 BLE
+bluetooth: {spec: 4.0 + LE}
 cameras:
 - {flash: 'LED', info: '16 MP'}
 - {flash: '', info: '2 MP'}
diff --git a/_data/devices/kltekdi.yml b/_data/devices/kltekdi.yml
index 379b2fc..6697be0 100644
--- a/_data/devices/kltekdi.yml
+++ b/_data/devices/kltekdi.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: 2800 mAh Li-ion battery, removable
-bluetooth: 4.0 BLE
+bluetooth: {spec: 4.0 + LE}
 cameras:
 - {flash: 'LED', info: '16 MP'}
 - {flash: '', info: '2 MP'}
diff --git a/_data/devices/kltekor.yml b/_data/devices/kltekor.yml
index 54e366e..d3d9835 100644
--- a/_data/devices/kltekor.yml
+++ b/_data/devices/kltekor.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: 2800 mAh Li-ion battery, removable
-bluetooth: 4.0 BLE
+bluetooth: {spec: 4.0 + LE}
 cameras:
 - {flash: 'LED', info: '16 MP'}
 - {flash: '', info: '2 MP'}
diff --git a/_data/devices/kltesprsports.yml b/_data/devices/kltesprsports.yml
index 93fc69f..ec17ecc 100644
--- a/_data/devices/kltesprsports.yml
+++ b/_data/devices/kltesprsports.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: 2800 mAh Li-ion battery, removable
-bluetooth: 4.0 BLE
+bluetooth: {spec: 4.0 + LE}
 cameras:
 - {flash: 'LED', info: '16 MP'}
 - {flash: '', info: '2 MP'}
diff --git a/_data/devices/ks01lte.yml b/_data/devices/ks01lte.yml
index cde092f..8b7626f 100644
--- a/_data/devices/ks01lte.yml
+++ b/_data/devices/ks01lte.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: 2600 mAh Li-ion, removable
-bluetooth: '4.0'
+bluetooth: {spec: '4.0'}
 cameras:
 - {flash: '', info: '13 MP, 1080p Full HD Video Recording @ 30fps'}
 - {flash: '', info: '2 MP, 1080p HD Video Recording @ 30fps'}
diff --git a/_data/devices/lentislte.yml b/_data/devices/lentislte.yml
index 3422a87..e12dcad 100644
--- a/_data/devices/lentislte.yml
+++ b/_data/devices/lentislte.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: 2800 mAh Li-ion battery, removable
-bluetooth: 4.0 BLE
+bluetooth: {spec: 4.0 + LE}
 cameras:
 - {flash: 'LED', info: '16 MP'}
 - {flash: '', info: '2 MP'}
diff --git a/_data/devices/lettuce.yml b/_data/devices/lettuce.yml
index 6b88cf2..a6b79ee 100644
--- a/_data/devices/lettuce.yml
+++ b/_data/devices/lettuce.yml
@@ -1,6 +1,6 @@
 architecture: arm64
 battery: 2230 mAh
-bluetooth: '4.0'
+bluetooth: {spec: '4.0'}
 cameras:
 - {flash: 'LED', info: '8 MP'}
 - {flash: '', info: '5 MP'}
diff --git a/_data/devices/libra.yml b/_data/devices/libra.yml
index ec937d5..d343f8c 100644
--- a/_data/devices/libra.yml
+++ b/_data/devices/libra.yml
@@ -1,6 +1,6 @@
 architecture: arm64
 battery: Non-removable Li-Ion 3080 mAh
-bluetooth: v4.1 with A2DP, LE
+bluetooth: {spec: '4.1', profiles: [A2DP]}
 cameras:
 - {flash: 'Dual LED (dual tone)', info: '13 MP, phase detection autofocus'}
 - {flash: '', info: '5 MP'}
diff --git a/_data/devices/lithium.yml b/_data/devices/lithium.yml
index dd5db9a..42b130b 100644
--- a/_data/devices/lithium.yml
+++ b/_data/devices/lithium.yml
@@ -1,6 +1,6 @@
 architecture: arm64
 battery: Non-removable Li-Po 4400 mAh
-bluetooth: v4.2 with A2DP, LE, aptX
+bluetooth: {spec: '4.2', profiles: [A2DP + aptX]}
 cameras:
 - {flash: 'Dual LED (dual tone)', info: '16 MP'}
 - {flash: '', info: '5 MP'}
diff --git a/_data/devices/ls990.yml b/_data/devices/ls990.yml
index 677ea74..6c3048a 100644
--- a/_data/devices/ls990.yml
+++ b/_data/devices/ls990.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Removable Li-Ion 3000 mAh
-bluetooth: v4.0 with A2DP, LE, aptX
+bluetooth: {spec: 4.0 + LE, profiles: [A2DP + aptX]}
 cameras:
 - {flash: 'Dual LED', info: '13 MP'}
 - {flash: '', info: '2.1 MP'}
diff --git a/_data/devices/ls997.yml b/_data/devices/ls997.yml
index a6031e5..a5a2936 100644
--- a/_data/devices/ls997.yml
+++ b/_data/devices/ls997.yml
@@ -1,6 +1,6 @@
 architecture: arm64
 battery: Removable Li-Ion 3200 mAh
-bluetooth: v4.2 with A2DP, LE, aptX HD
+bluetooth: {spec: '4.2', profiles: [A2DP + aptX HD]}
 cameras:
 - {flash: 'LED', info: '16 MP'}
 - {flash: 'LED', info: '8 MP'}
diff --git a/_data/devices/lt02ltespr.yml b/_data/devices/lt02ltespr.yml
index 3419a46..45e65e1 100644
--- a/_data/devices/lt02ltespr.yml
+++ b/_data/devices/lt02ltespr.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Non-removable Li-Ion 4000 mAh
-bluetooth: v4.0 with A2DP, LE
+bluetooth: {spec: 4.0 + LE, profiles: [A2DP]}
 cameras:
 - {flash: '', info: '3 MP'}
 - {flash: '', info: '1.3 MP'}
diff --git a/_data/devices/lt03lte.yml b/_data/devices/lt03lte.yml
index b92f307..29bbc8a 100644
--- a/_data/devices/lt03lte.yml
+++ b/_data/devices/lt03lte.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: 8220 mAh
-bluetooth: '4.0'
+bluetooth: {spec: '4.0'}
 cameras:
 - {flash: 'LED', info: '8 MP'}
 - {flash: '', info: '2.1 MP'}
diff --git a/_data/devices/lux.yml b/_data/devices/lux.yml
index 12afc76..4eb7e44 100644
--- a/_data/devices/lux.yml
+++ b/_data/devices/lux.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: 3630 mAh
-bluetooth: 4.0 LE
+bluetooth: {spec: 4.0 + LE}
 cameras:
 - {flash: 'Dual LED', info: '21 MP'}
 - {flash: '', info: '5 MP'}
diff --git a/_data/devices/m216.yml b/_data/devices/m216.yml
index 4e09e70..49b6161 100644
--- a/_data/devices/m216.yml
+++ b/_data/devices/m216.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Removable Li-Ion 2300 mAh
-bluetooth: 4.1
+bluetooth: {spec: '4.1'}
 cameras:
 - {flash: 'LED', info: '13 MP(LTE) or 8 MP(3G)'}
 - {flash: '', info: '5 MP or 8 MP'}
diff --git a/_data/devices/m7.yml b/_data/devices/m7.yml
index 04ddc66..3543df0 100644
--- a/_data/devices/m7.yml
+++ b/_data/devices/m7.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: 2300 mAh
-bluetooth: '4.0'
+bluetooth: {spec: '4.0'}
 cameras:
 - {flash: 'LED', info: '4 UP'}
 - {flash: '', info: '2.1 MP'}
diff --git a/_data/devices/m7vzw.yml b/_data/devices/m7vzw.yml
index d161b09..c3503fd 100644
--- a/_data/devices/m7vzw.yml
+++ b/_data/devices/m7vzw.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: 2300 mAh
-bluetooth: '4.0'
+bluetooth: {spec: '4.0'}
 cameras:
 - {flash: 'LED', info: '4 UP'}
 - {flash: '', info: '2.1 MP'}
diff --git a/_data/devices/m8.yml b/_data/devices/m8.yml
index b5b3ad6..fb5ab95 100644
--- a/_data/devices/m8.yml
+++ b/_data/devices/m8.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Non-removable Li-Po 2600 mAh
-bluetooth: v4.0 with A2DP
+bluetooth: {spec: '4.0', profiles: [A2DP]}
 cameras:
 - {flash: 'Dual LED', info: 'Dual 4 MP'}
 - {flash: '', info: '5 MP'}
diff --git a/_data/devices/m8d.yml b/_data/devices/m8d.yml
index 589583d..f850e9b 100644
--- a/_data/devices/m8d.yml
+++ b/_data/devices/m8d.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Non-removable Li-Po 2600 mAh
-bluetooth: v4.0 with A2DP
+bluetooth: {spec: '4.0', profiles: [A2DP]}
 cameras:
 - {flash: 'LED', info: 'Dual 4 MP'}
 - {flash: '', info: '5 MP'}
diff --git a/_data/devices/maguro.yml b/_data/devices/maguro.yml
index d6cd24d..72aefc1 100644
--- a/_data/devices/maguro.yml
+++ b/_data/devices/maguro.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Removable Li-Ion 1750 mAh
-bluetooth: v4.0 with A2DP
+bluetooth: {spec: '4.0', profiles: [A2DP]}
 cameras:
 - {flash: 'LED', info: '5 MP'}
 - {flash: '', info: '1.3 MP'}
diff --git a/_data/devices/mako.yml b/_data/devices/mako.yml
index 0a9f21b..75a7880 100644
--- a/_data/devices/mako.yml
+++ b/_data/devices/mako.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: 3.8 V 2100 mAh, 8 Wh rechargable Lithium polymer non-removable battery
-bluetooth: '4.0'
+bluetooth: {spec: '4.0'}
 cameras:
 - {flash: 'LED', info: '8 MP'}
 - {flash: '', info: '1.3 MP'}
diff --git a/_data/devices/manta.yml b/_data/devices/manta.yml
index 1074e73..b986c54 100644
--- a/_data/devices/manta.yml
+++ b/_data/devices/manta.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Non-removable Li-Po 9000 mAh
-bluetooth: v3.0 with A2DP
+bluetooth: {spec: '3.0', profiles: [A2DP]}
 cameras:
 - {flash: 'LED', info: '5 MP'}
 - {flash: '', info: '1.9 MP'}
diff --git a/_data/devices/maserati.yml b/_data/devices/maserati.yml
index 71ea341..e7462df 100644
--- a/_data/devices/maserati.yml
+++ b/_data/devices/maserati.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: 1785 mAh
-bluetooth: ' v4.0 with LE+EDR'
+bluetooth: {spec: 4.0 + LE}
 cameras:
 - {flash: 'LED', info: '8 MP'}
 - {flash: '', info: '1.3 MP'}
diff --git a/_data/devices/merlin.yml b/_data/devices/merlin.yml
index b500c81..cb07a41 100644
--- a/_data/devices/merlin.yml
+++ b/_data/devices/merlin.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: 2470 mAh
-bluetooth: v4.0 with A2DP, LE
+bluetooth: {spec: 4.0 + LE, profiles: [A2DP]}
 cameras:
 - {flash: 'Dual LED', info: '13 MP'}
 - {flash: '', info: '5 MP'}
diff --git a/_data/devices/mido.yml b/_data/devices/mido.yml
index 4fdd4d0..152f454 100644
--- a/_data/devices/mido.yml
+++ b/_data/devices/mido.yml
@@ -1,6 +1,6 @@
 architecture: arm64
 battery: Non-removable Li-Po 4100 mAh
-bluetooth: v4.1 with A2DP, LE
+bluetooth: {spec: '4.1', profiles: [A2DP]}
 cameras:
 - {flash: 'LED (dual-tone)', info: '13 MP'}
 - {flash: '', info: '5 MP'}
diff --git a/_data/devices/mint.yml b/_data/devices/mint.yml
index 0ef0733..3e0ea45 100644
--- a/_data/devices/mint.yml
+++ b/_data/devices/mint.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Non-removable 1780 mAh
-bluetooth: v4.0 with A2DP
+bluetooth: {spec: '4.0', profiles: [A2DP]}
 cameras:
 - {flash: 'LED', info: '13 MP f/2.4'}
 - {flash: '', info: '1.3 MP'}
diff --git a/_data/devices/mondrianwifi.yml b/_data/devices/mondrianwifi.yml
index a1117be..1c1abcc 100644
--- a/_data/devices/mondrianwifi.yml
+++ b/_data/devices/mondrianwifi.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery:
-bluetooth: v4.0 LE
+bluetooth: {spec: 4.0 + LE}
 cameras:
 - {flash: 'LED', info: '8 MP'}
 - {flash: '', info: '2 MP'}
diff --git a/_data/devices/mt2.yml b/_data/devices/mt2.yml
index ba76885..63cca35 100644
--- a/_data/devices/mt2.yml
+++ b/_data/devices/mt2.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Non-removable Li-Ion 4050 mAh
-bluetooth: '4.0'
+bluetooth: {spec: '4.0'}
 cameras:
 - {flash: 'LED', info: '13 MP'}
 - {flash: '', info: '5 MP'}
diff --git a/_data/devices/n1awifi.yml b/_data/devices/n1awifi.yml
index 6201288..3b2fc87 100644
--- a/_data/devices/n1awifi.yml
+++ b/_data/devices/n1awifi.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Non-removable Li-Po 8220 mAh
-bluetooth: 4.0 A2DP
+bluetooth: {spec: '4.0', profiles: [A2DP]}
 cameras:
 - {flash: 'LED', info: '8 MP'}
 - {flash: '', info: '2 MP'}
diff --git a/_data/devices/n2awifi.yml b/_data/devices/n2awifi.yml
index bf07e93..0809912 100644
--- a/_data/devices/n2awifi.yml
+++ b/_data/devices/n2awifi.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Non-removable Li-Ion 8220 mAh
-bluetooth: v4.0 with A2DP
+bluetooth: {spec: '4.0', profiles: [A2DP]}
 cameras:
 - {flash: 'LED', info: '8 MP'}
 - {flash: '', info: '2.1 MP'}
diff --git a/_data/devices/n3.yml b/_data/devices/n3.yml
index ae83936..594713b 100644
--- a/_data/devices/n3.yml
+++ b/_data/devices/n3.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Non-removable 3000 mAh Li-Po with VOOC Flash Charge
-bluetooth: '4.0'
+bluetooth: {spec: '4.0'}
 cameras:
 - {flash: Dual-mode LED, info: 'Motorized Rotation, 16MP OmniVision OV16825 f/2.2'}
 carrier:
diff --git a/_data/devices/n5100.yml b/_data/devices/n5100.yml
index 3e6c0da..a73c165 100644
--- a/_data/devices/n5100.yml
+++ b/_data/devices/n5100.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Non-removable Li-Ion 4600 mAh
-bluetooth: v4.0 with A2DP
+bluetooth: {spec: '4.0', profiles: [A2DP]}
 cameras:
 - {flash: '', info: '5 MP, autofocus'}
 - {flash: '', info: '1.3 MP'}
diff --git a/_data/devices/n5110.yml b/_data/devices/n5110.yml
index 5fbde25..dba1114 100644
--- a/_data/devices/n5110.yml
+++ b/_data/devices/n5110.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Non-removable Li-Ion 4600 mAh
-bluetooth: v4.0 with A2DP
+bluetooth: {spec: '4.0', profiles: [A2DP]}
 cameras:
 - {flash: '', info: '5 MP, autofocus'}
 - {flash: '', info: '1.3 MP'}
diff --git a/_data/devices/n5120.yml b/_data/devices/n5120.yml
index dbe1631..795b98b 100644
--- a/_data/devices/n5120.yml
+++ b/_data/devices/n5120.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Non-removable Li-Ion 4600 mAh
-bluetooth: v4.0 with A2DP
+bluetooth: {spec: '4.0', profiles: [A2DP]}
 cameras:
 - {flash: '', info: '5 MP, autofocus'}
 - {flash: '', info: '1.3 MP'}
diff --git a/_data/devices/natrium.yml b/_data/devices/natrium.yml
index 36c10cc..fa22887 100644
--- a/_data/devices/natrium.yml
+++ b/_data/devices/natrium.yml
@@ -1,6 +1,6 @@
 architecture: arm64
 battery: Non-removable Li-Po 3800 mAh
-bluetooth: v4.2 with A2DP, LE, aptX
+bluetooth: {spec: '4.2', profiles: [A2DP + aptX]}
 cameras:
 - {flash: 'Dual LED (dual tone)', info: '16 MP'}
 - {flash: 'Dual LED (dual tone)', info: 'Monochromatic 16 MP'}
diff --git a/_data/devices/nicki.yml b/_data/devices/nicki.yml
index 3f16358..a20d8d6 100644
--- a/_data/devices/nicki.yml
+++ b/_data/devices/nicki.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Removable Li-Po 1750 mAh
-bluetooth: v4.0 + A2DP
+bluetooth: {spec: '4.0', profiles: [A2DP]}
 cameras:
 - {flash: '', info: '5 MP'}
 - {flash: '', info: 'VGA'}
diff --git a/_data/devices/nx512j.yml b/_data/devices/nx512j.yml
index 514b9c7..d4ddd3a 100644
--- a/_data/devices/nx512j.yml
+++ b/_data/devices/nx512j.yml
@@ -1,6 +1,6 @@
 architecture: arm64
 battery: Non-removable Li-Ion 2900 mAh
-bluetooth: v4.0
+bluetooth: {spec: '4.0'}
 cameras:
 - {flash: 'LED', info: '15.9 MP'}
 - {flash: 'LED', info: '8 MP'}
diff --git a/_data/devices/odin.yml b/_data/devices/odin.yml
index 3e49420..8b4cef0 100644
--- a/_data/devices/odin.yml
+++ b/_data/devices/odin.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Non-removable 2370 mAh
-bluetooth: v4.0 with A2DP, LE
+bluetooth: {spec: 4.0 + LE, profiles: [A2DP]}
 cameras:
 - {flash: 'LED', info: '13MP f/2.4'}
 - {flash: '', info: '2MP'}
diff --git a/_data/devices/oneplus2.yml b/_data/devices/oneplus2.yml
index 4d98f27..a588420 100644
--- a/_data/devices/oneplus2.yml
+++ b/_data/devices/oneplus2.yml
@@ -1,6 +1,6 @@
 architecture: arm64
 battery: 3300 mAh
-bluetooth: '4.1'
+bluetooth: {spec: '4.1'}
 cameras:
 - {flash: 'Dual LED', info: '13 MP  F/2.0'}
 - {flash: '', info: '5 MP F/2.4'}
diff --git a/_data/devices/oneplus3.yml b/_data/devices/oneplus3.yml
index 61ef83c..abc038f 100644
--- a/_data/devices/oneplus3.yml
+++ b/_data/devices/oneplus3.yml
@@ -1,6 +1,6 @@
 architecture: arm64
 battery: Non-removable Li-Ion 3000 mAh (3) / 3400 mA (3T)
-bluetooth: v4.2 with A2DP, LE
+bluetooth: {spec: '4.2', profiles: [A2DP]}
 cameras:
 - {flash: 'LED', info: '16 MP'}
 - {flash: '', info: 8MP (3) / 16MP (3T)}
diff --git a/_data/devices/onyx.yml b/_data/devices/onyx.yml
index bf0c336..cc24b29 100644
--- a/_data/devices/onyx.yml
+++ b/_data/devices/onyx.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Non-removable Li-Po 2525 mAh
-bluetooth: v4.0 with A2DP
+bluetooth: {spec: '4.0', profiles: [A2DP]}
 cameras:
 - {flash: 'LED', info: '13 MP'}
 - {flash: '', info: '8 MP'}
diff --git a/_data/devices/osprey.yml b/_data/devices/osprey.yml
index e164c1e..2eaafb5 100644
--- a/_data/devices/osprey.yml
+++ b/_data/devices/osprey.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: 2470 mAh
-bluetooth: v4.0 with A2DP, LE
+bluetooth: {spec: 4.0 + LE, profiles: [A2DP]}
 cameras:
 - {flash: 'Dual LED', info: '13 MP'}
 - {flash: '', info: '5 MP'}
diff --git a/_data/devices/otus.yml b/_data/devices/otus.yml
index 5fee696..6e7dab9 100644
--- a/_data/devices/otus.yml
+++ b/_data/devices/otus.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: 2390 mAh
-bluetooth: v4.0
+bluetooth: {spec: '4.0'}
 cameras:
 - {flash: '', info: '5 MP'}
 - {flash: '', info: 'VGA'}
diff --git a/_data/devices/paella.yml b/_data/devices/paella.yml
index 238c9d7..e0f29ca 100644
--- a/_data/devices/paella.yml
+++ b/_data/devices/paella.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Non-removable Li-Po 2900 mAh
-bluetooth: v4.0
+bluetooth: {spec: '4.0'}
 cameras:
 - {flash: 'Dual LED', info: '13 MP'}
 - {flash: 'LED', info: '5 MP'}
diff --git a/_data/devices/peach.yml b/_data/devices/peach.yml
index e6e3c9a..a0624f5 100644
--- a/_data/devices/peach.yml
+++ b/_data/devices/peach.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Removable Li-Ion 2100 mAh
-bluetooth: v4.0 with A2DP, LE
+bluetooth: {spec: 4.0 + LE, profiles: [A2DP]}
 cameras:
 - {flash: 'Dual LED', info: '8 MP'}
 - {flash: '', info: '2 MP'}
diff --git a/_data/devices/peregrine.yml b/_data/devices/peregrine.yml
index aaa67e9..b1ecce7 100644
--- a/_data/devices/peregrine.yml
+++ b/_data/devices/peregrine.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: 2070 mAh
-bluetooth: v4.0 with A2DP, LE
+bluetooth: {spec: 4.0 + LE, profiles: [A2DP]}
 cameras:
 - {flash: 'LED', info: '5 MP'}
 - {flash: '', info: '1.3 MP'}
diff --git a/_data/devices/piccolo.yml b/_data/devices/piccolo.yml
index 734c403..317fee5 100644
--- a/_data/devices/piccolo.yml
+++ b/_data/devices/piccolo.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Non-removable Li-Po 3120 mAh
-bluetooth: v4.0
+bluetooth: {spec: '4.0'}
 cameras:
 - {flash: 'Dual LED', info: '13 MP'}
 - {flash: '', info: '5 MP'}
diff --git a/_data/devices/pme.yml b/_data/devices/pme.yml
index 2bab7a5..b5bfd4f 100644
--- a/_data/devices/pme.yml
+++ b/_data/devices/pme.yml
@@ -1,6 +1,6 @@
 architecture: arm64
 battery: Non-removable Li-Ion 3000 mAh
-bluetooth: v4.2 with A2DP, aptX
+bluetooth: {spec: '4.2', profiles: [A2DP + aptX]}
 cameras:
 - {flash: 'Dual LED', info: '12 MP , f/1.8, 26mm'}
 - {flash: '', info: '5 MP, f/1.8, 23mm'}
diff --git a/_data/devices/pollux.yml b/_data/devices/pollux.yml
index d71d670..7753211 100644
--- a/_data/devices/pollux.yml
+++ b/_data/devices/pollux.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Non-removable 6000 mAh
-bluetooth: v4.0 with A2DP, LE
+bluetooth: {spec: 4.0 + LE, profiles: [A2DP]}
 cameras:
 - {flash: 'LED', info: '8.1MP f/2.4'}
 - {flash: '', info: '2.2MP'}
diff --git a/_data/devices/pollux_windy.yml b/_data/devices/pollux_windy.yml
index 67b9282..9216c8f 100644
--- a/_data/devices/pollux_windy.yml
+++ b/_data/devices/pollux_windy.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Non-removable 6000 mAh
-bluetooth: v4.0 with A2DP, LE
+bluetooth: {spec: 4.0 + LE, profiles: [A2DP]}
 cameras:
 - {flash: 'LED', info: '8.1MP f/2.4'}
 - {flash: '', info: '2.2MP'}
diff --git a/_data/devices/r5.yml b/_data/devices/r5.yml
index 0101f58..da3aab4 100644
--- a/_data/devices/r5.yml
+++ b/_data/devices/r5.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Non-removable Li-Po 2000 mAh
-bluetooth: '4.0'
+bluetooth: {spec: '4.0'}
 cameras:
 - {flash: 'LED', info: '13 MP'}
 - {flash: '', info: '5 MP'}
diff --git a/_data/devices/r7plus.yml b/_data/devices/r7plus.yml
index ad27da0..51221d9 100644
--- a/_data/devices/r7plus.yml
+++ b/_data/devices/r7plus.yml
@@ -1,6 +1,6 @@
 architecture: arm64
 battery: Non-removable Li-Po 4100 mAh
-bluetooth: '4.0'
+bluetooth: {spec: '4.0'}
 cameras:
 - {flash: 'Dual LED', info: '13 MP'}
 - {flash: '', info: '8 MP'}
diff --git a/_data/devices/r7sf.yml b/_data/devices/r7sf.yml
index 6f7c52a..f383447 100644
--- a/_data/devices/r7sf.yml
+++ b/_data/devices/r7sf.yml
@@ -1,6 +1,6 @@
 architecture: arm64
 battery: Non-removable Li-Po 3070 mAh
-bluetooth: '4.0'
+bluetooth: {spec: '4.0'}
 cameras:
 - {flash: 'LED', info: '13 MP'}
 - {flash: '', info: '8 MP'}
diff --git a/_data/devices/roth.yml b/_data/devices/roth.yml
index 422378f..cd87295 100644
--- a/_data/devices/roth.yml
+++ b/_data/devices/roth.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: 7350 mAh
-bluetooth: '3.0'
+bluetooth: {spec: '3.0'}
 cameras: []
 carrier:
 channels: [weekly]
diff --git a/_data/devices/s2.yml b/_data/devices/s2.yml
index 519c9e7..435af61 100644
--- a/_data/devices/s2.yml
+++ b/_data/devices/s2.yml
@@ -1,6 +1,6 @@
 architecture: arm64
 battery: 3000 mAh (non-removable)
-bluetooth: v4.1 with A2DP, BLE
+bluetooth: {spec: '4.1', profiles: [A2DP]}
 cameras:
 - {flash: 'Dual LED (dual tone)', info: '16 MP'}
 - {flash: '', info: '8 MP'}
diff --git a/_data/devices/seed.yml b/_data/devices/seed.yml
index 56a08a2..679c5fe 100644
--- a/_data/devices/seed.yml
+++ b/_data/devices/seed.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: 2500 mAh
-bluetooth: '4.0'
+bluetooth: {spec: '4.0'}
 cameras:
 - {flash: 'LED', info: '13 MP'}
 - {flash: '', info: '5 MP'}
diff --git a/_data/devices/serrano3gxx.yml b/_data/devices/serrano3gxx.yml
index d20f6fd..3ed8be2 100644
--- a/_data/devices/serrano3gxx.yml
+++ b/_data/devices/serrano3gxx.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Removable Li-Ion 1900 mAh
-bluetooth: v4.0 with A2DP, LE, EDR
+bluetooth: {spec: 4.0 + LE, profiles: [A2DP]}
 cameras:
 - {flash: 'LED', info: '8 MP, 1080p Full HD Video Recording @ 30fps'}
 - {flash: '', info: '1.9 MP, 720p HD Video Recording @ 30fps'}
diff --git a/_data/devices/serranodsdd.yml b/_data/devices/serranodsdd.yml
index 81ec8d5..5d90df8 100644
--- a/_data/devices/serranodsdd.yml
+++ b/_data/devices/serranodsdd.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Removable Li-Ion 1900 mAh
-bluetooth: v4.0 with A2DP, LE, EDR
+bluetooth: {spec: 4.0 + LE, profiles: [A2DP]}
 cameras:
 - {flash: 'LED', info: '8 MP, 1080p Full HD Video Recording @ 30fps'}
 - {flash: '', info: '1.9 MP, 720p HD Video Recording @ 30fps'}
diff --git a/_data/devices/serranoltexx.yml b/_data/devices/serranoltexx.yml
index e67f4d4..6478753 100644
--- a/_data/devices/serranoltexx.yml
+++ b/_data/devices/serranoltexx.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Removable Li-Ion 1900 mAh
-bluetooth: v4.0 with A2DP, LE, EDR
+bluetooth: {spec: 4.0 + LE, profiles: [A2DP]}
 cameras:
 - {flash: 'LED', info: '8 MP, 1080p Full HD Video Recording @ 30fps'}
 - {flash: '', info: '1.9 MP, 720p HD Video Recording @ 30fps'}
diff --git a/_data/devices/shamu.yml b/_data/devices/shamu.yml
index 122562c..ea6ae94 100644
--- a/_data/devices/shamu.yml
+++ b/_data/devices/shamu.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Non-removable Li-Po 3220 mAh
-bluetooth: v4.1 LE
+bluetooth: {spec: '4.1'}
 cameras:
 - {flash: 'Dual LED', info: '13 MP'}
 - {flash: '', info: '2.1 MP '}
diff --git a/_data/devices/shieldtablet.yml b/_data/devices/shieldtablet.yml
index 45da857..be7ff38 100644
--- a/_data/devices/shieldtablet.yml
+++ b/_data/devices/shieldtablet.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: 5197 mAh
-bluetooth: 4.0 LE
+bluetooth: {spec: 4.0 + LE}
 cameras:
 - {flash: '', info: '5 MP'}
 - {flash: '', info: '5 MP'}
diff --git a/_data/devices/spyder.yml b/_data/devices/spyder.yml
index a054ed4..b71fcc5 100644
--- a/_data/devices/spyder.yml
+++ b/_data/devices/spyder.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: 1780 mAh
-bluetooth: v4.0 with LE+EDR
+bluetooth: {spec: 4.0 + LE}
 cameras:
 - {flash: 'LED', info: '8 MP'}
 - {flash: '', info: '2 MP'}
diff --git a/_data/devices/surnia.yml b/_data/devices/surnia.yml
index 56ff61d..ee9bb56 100644
--- a/_data/devices/surnia.yml
+++ b/_data/devices/surnia.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: 2390 mAh
-bluetooth: v4.0
+bluetooth: {spec: '4.0'}
 cameras:
 - {flash: '', info: '5 MP'}
 - {flash: '', info: 'VGA'}
diff --git a/_data/devices/suzuran.yml b/_data/devices/suzuran.yml
index 942dde1..2a94c75 100644
--- a/_data/devices/suzuran.yml
+++ b/_data/devices/suzuran.yml
@@ -1,6 +1,6 @@
 architecture: arm64
 battery: Non-removable Li-Ion 2700 mAh
-bluetooth: v4.1 with A2DP, aptX
+bluetooth: {spec: '4.1', profiles: [A2DP + aptX]}
 cameras:
 - {flash: 'LED', info: '23 MP f/2.0'}
 - {flash: '', info: '5.1 MP f/2.4'}
diff --git a/_data/devices/t0lte.yml b/_data/devices/t0lte.yml
index 2baa59d..5f35972 100644
--- a/_data/devices/t0lte.yml
+++ b/_data/devices/t0lte.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Removable Li-Ion 3100 mAh
-bluetooth: '4.0'
+bluetooth: {spec: '4.0'}
 cameras:
 - {flash: 'LED', info: '8 MP'}
 - {flash: '', info: '1.9 MP'}
diff --git a/_data/devices/t6.yml b/_data/devices/t6.yml
index 3776942..eb4fd2a 100644
--- a/_data/devices/t6.yml
+++ b/_data/devices/t6.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Non-removable 3300 mAh
-bluetooth: v4.0 with A2DP, aptX
+bluetooth: {spec: '4.0', profiles: [A2DP + aptX]}
 cameras:
 - {flash: 'LED', info: '4 UP'}
 - {flash: '', info: '2.1 MP'}
diff --git a/_data/devices/t6vzw.yml b/_data/devices/t6vzw.yml
index 4042497..9253467 100644
--- a/_data/devices/t6vzw.yml
+++ b/_data/devices/t6vzw.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Non-removable 3300 mAh
-bluetooth: v4.0 with A2DP, aptX
+bluetooth: {spec: '4.0', profiles: [A2DP + aptX]}
 cameras:
 - {flash: 'LED', info: '4 UP'}
 - {flash: '', info: '2.1 MP'}
diff --git a/_data/devices/taoshan.yml b/_data/devices/taoshan.yml
index 1802ce3..a526304 100644
--- a/_data/devices/taoshan.yml
+++ b/_data/devices/taoshan.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Removable Li-Ion 1750 mAh
-bluetooth: v4.0 with A2DP
+bluetooth: {spec: '4.0', profiles: [A2DP]}
 cameras:
 - {flash: 'LED', info: '8 MP f/2.4'}
 - {flash: '', info: 'VGA'}
diff --git a/_data/devices/targa.yml b/_data/devices/targa.yml
index 245c73e..2499ac5 100644
--- a/_data/devices/targa.yml
+++ b/_data/devices/targa.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: 1735 mAh
-bluetooth: v2.1 with A2DP, EDR
+bluetooth: {spec: 2.1 + EDR, profiles: [A2DP]}
 cameras:
 - {flash: 'LED', info: '8 MP'}
 - {flash: '', info: 'VGA'}
diff --git a/_data/devices/tenshi.yml b/_data/devices/tenshi.yml
index bbc6373..ee0a0c9 100644
--- a/_data/devices/tenshi.yml
+++ b/_data/devices/tenshi.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Non-removable Li-Po 3050 mAh
-bluetooth: v4.0
+bluetooth: {spec: '4.0'}
 cameras:
 - {flash: 'Dual LED', info: '16 MP'}
 - {flash: '', info: '5 MP'}
diff --git a/_data/devices/thea.yml b/_data/devices/thea.yml
index 16e7bb3..107a751 100644
--- a/_data/devices/thea.yml
+++ b/_data/devices/thea.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: 2390 mAh
-bluetooth: v4.0 with A2DP, LE
+bluetooth: {spec: 4.0 + LE, profiles: [A2DP]}
 cameras:
 - {flash: 'LED', info: '8 MP'}
 - {flash: '', info: '2 MP'}
diff --git a/_data/devices/titan.yml b/_data/devices/titan.yml
index 7d9039d..a52d7cf 100644
--- a/_data/devices/titan.yml
+++ b/_data/devices/titan.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: 2070 mAh
-bluetooth: v4.0 with A2DP, LE
+bluetooth: {spec: 4.0 + LE, profiles: [A2DP]}
 cameras:
 - {flash: 'LED', info: '8 MP'}
 - {flash: '', info: '2 MP'}
diff --git a/_data/devices/tomato.yml b/_data/devices/tomato.yml
index c9f7da3..c9780a8 100644
--- a/_data/devices/tomato.yml
+++ b/_data/devices/tomato.yml
@@ -1,6 +1,6 @@
 architecture: arm64
 battery: 2500 mAh
-bluetooth: v4.1
+bluetooth: {spec: '4.1'}
 cameras:
 - {flash: 'LED', info: '13 MP F/2.2'}
 - {flash: '', info: '5 MP'}
diff --git a/_data/devices/toro.yml b/_data/devices/toro.yml
index b0a4ec5..4e3028d 100644
--- a/_data/devices/toro.yml
+++ b/_data/devices/toro.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Removable Li-Ion 1850 mAh
-bluetooth: v4.0 with A2DP
+bluetooth: {spec: '4.0', profiles: [A2DP]}
 cameras:
 - {flash: 'LED', info: '5 MP'}
 - {flash: '', info: '1.3 MP'}
diff --git a/_data/devices/toroplus.yml b/_data/devices/toroplus.yml
index 1fb9064..8d8df42 100644
--- a/_data/devices/toroplus.yml
+++ b/_data/devices/toroplus.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Removable Li-Ion 1850 mAh
-bluetooth: v4.0 with A2DP
+bluetooth: {spec: '4.0', profiles: [A2DP]}
 cameras:
 - {flash: 'LED', info: '5 MP'}
 - {flash: '', info: '1.3 MP'}
diff --git a/_data/devices/tsubasa.yml b/_data/devices/tsubasa.yml
index a448b75..f476261 100644
--- a/_data/devices/tsubasa.yml
+++ b/_data/devices/tsubasa.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Removable 1700 mAh
-bluetooth: v4.0 with A2DP
+bluetooth: {spec: '4.0', profiles: [A2DP]}
 cameras:
 - {flash: 'LED', info: '13 MP f/2.4'}
 - {flash: '', info: '1.3 MP'}
diff --git a/_data/devices/umts_spyder.yml b/_data/devices/umts_spyder.yml
index 1b7952e..c276ee5 100644
--- a/_data/devices/umts_spyder.yml
+++ b/_data/devices/umts_spyder.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: 1780 mAh / 3300 mAh (MAXX)
-bluetooth: v4.0 with LE+EDR
+bluetooth: {spec: 4.0 + LE}
 cameras:
 - {flash: 'LED', info: '8 MP'}
 - {flash: '', info: '1.3 MP'}
diff --git a/_data/devices/us996.yml b/_data/devices/us996.yml
index d398d35..f578396 100644
--- a/_data/devices/us996.yml
+++ b/_data/devices/us996.yml
@@ -1,6 +1,6 @@
 architecture: arm64
 battery: Removable Li-Ion 3200 mAh
-bluetooth: v4.2 with A2DP, LE, aptX HD
+bluetooth: {spec: '4.2', profiles: [A2DP + aptX HD]}
 cameras:
 - {flash: 'LED', info: '16 MP'}
 - {flash: 'LED', info: '8 MP'}
diff --git a/_data/devices/us997.yml b/_data/devices/us997.yml
index 2348ee6..1ce2845 100644
--- a/_data/devices/us997.yml
+++ b/_data/devices/us997.yml
@@ -1,6 +1,6 @@
 architecture: arm64
 battery: Removable Li-Ion 3300 mAh
-bluetooth: v4.2 with A2DP, LE, aptX HD
+bluetooth: {spec: '4.2', profiles: [A2DP + aptX HD]}
 cameras:
 - {flash: 'LED', info: '13 MP'}
 - {flash: 'LED', info: '13 MP'}
diff --git a/_data/devices/v1awifi.yml b/_data/devices/v1awifi.yml
index 622185f..8abc7a1 100644
--- a/_data/devices/v1awifi.yml
+++ b/_data/devices/v1awifi.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Non-removable Li-Ion 9500 mAh
-bluetooth: 4.0 A2DP
+bluetooth: {spec: '4.0', profiles: [A2DP]}
 cameras:
 - {flash: 'LED', info: '8 MP'}
 - {flash: '', info: '2 MP'}
diff --git a/_data/devices/v400.yml b/_data/devices/v400.yml
index 24129cd..9c41cb9 100644
--- a/_data/devices/v400.yml
+++ b/_data/devices/v400.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Non-removable Li-Ion 4000 mAh
-bluetooth: v4.0
+bluetooth: {spec: '4.0'}
 cameras:
 - {flash: '', info: '3.15 MP'}
 - {flash: '', info: '1.3 MP'}
diff --git a/_data/devices/v410.yml b/_data/devices/v410.yml
index 05e349e..3cacda5 100644
--- a/_data/devices/v410.yml
+++ b/_data/devices/v410.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Non-removable Li-Ion 4000 mAh
-bluetooth: v4.0
+bluetooth: {spec: '4.0'}
 cameras:
 - {flash: '', info: '5 MP'}
 - {flash: '', info: '1.3 MP'}
diff --git a/_data/devices/v480.yml b/_data/devices/v480.yml
index 103545e..03a50b0 100644
--- a/_data/devices/v480.yml
+++ b/_data/devices/v480.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Non-removable Li-Ion 4200 mAh
-bluetooth: v4.0, LE
+bluetooth: {spec: 4.0 + LE}
 cameras:
 - {flash: '', info: '5 MP'}
 - {flash: '', info: '1.3 MP'}
diff --git a/_data/devices/v500.yml b/_data/devices/v500.yml
index a4e6933..601d2ed 100644
--- a/_data/devices/v500.yml
+++ b/_data/devices/v500.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Non-removable Li-Po 4600 mAh
-bluetooth: v4.0 LE
+bluetooth: {spec: 4.0 + LE}
 cameras:
 - {flash: '', info: '5 MP'}
 - {flash: '', info: '1.3 MP'}
diff --git a/_data/devices/v521.yml b/_data/devices/v521.yml
index b8031db..698ca6b 100644
--- a/_data/devices/v521.yml
+++ b/_data/devices/v521.yml
@@ -1,6 +1,6 @@
 architecture: arm64
 battery: Non-removable Li-Ion 4650 mAh
-bluetooth: v4.2 with A2DP, LE, aptX HD
+bluetooth: {spec: '4.2', profiles: [A2DP + aptX HD]}
 cameras:
 - {flash: '', info: '5 MP'}
 - {flash: '', info: '5 MP'}
diff --git a/_data/devices/vegetalte.yml b/_data/devices/vegetalte.yml
index 9cc8357..cbc8c05 100644
--- a/_data/devices/vegetalte.yml
+++ b/_data/devices/vegetalte.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Non-removable Li-Po 2850 mAh
-bluetooth: v4.0
+bluetooth: {spec: '4.0'}
 cameras:
 - {flash: 'Dual LED', info: '13 MP'}
 - {flash: '', info: '5 MP'}
diff --git a/_data/devices/victara.yml b/_data/devices/victara.yml
index c7662d6..e77c095 100644
--- a/_data/devices/victara.yml
+++ b/_data/devices/victara.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: 2300 mAh (non-removable)
-bluetooth: v4.0 with AD2P, EDR, LE
+bluetooth: {spec: 4.0 + LE, profiles: [A2DP]}
 cameras:
 - {flash: 'Dual LED', info: '13 MP, 4128 x 3096 pixels, autofocus'}
 - {flash: '', info: '2 MP, 1080p'}
diff --git a/_data/devices/vs985.yml b/_data/devices/vs985.yml
index 7797960..89070f0 100644
--- a/_data/devices/vs985.yml
+++ b/_data/devices/vs985.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Removable Li-Ion 3000 mAh
-bluetooth: v4.0 with A2DP, LE, aptX
+bluetooth: {spec: 4.0 + LE, profiles: [A2DP + aptX]}
 cameras:
 - {flash: 'Dual LED', info: '13 MP'}
 - {flash: '', info: '2.1 MP'}
diff --git a/_data/devices/vs995.yml b/_data/devices/vs995.yml
index 40f25a7..d470907 100644
--- a/_data/devices/vs995.yml
+++ b/_data/devices/vs995.yml
@@ -1,6 +1,6 @@
 architecture: arm64
 battery: Removable Li-Ion 3200 mAh
-bluetooth: v4.2 with A2DP, LE, aptX HD
+bluetooth: {spec: '4.2', profiles: [A2DP + aptX HD]}
 cameras:
 - {flash: 'LED', info: '16 MP'}
 - {flash: 'LED', info: '8 MP'}
diff --git a/_data/devices/w5.yml b/_data/devices/w5.yml
index 6da3427..3b54a60 100644
--- a/_data/devices/w5.yml
+++ b/_data/devices/w5.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: 2040 mAh
-bluetooth: '4.0'
+bluetooth: {spec: '4.0'}
 cameras:
 - {flash: 'LED', info: '5 MP'}
 - {flash: '', info: '0.3 MP'}
diff --git a/_data/devices/w7.yml b/_data/devices/w7.yml
index b41808d..2f5b638 100644
--- a/_data/devices/w7.yml
+++ b/_data/devices/w7.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Removable Li-Ion 2540 mAh
-bluetooth: v4.0 with A2DP, LE, aptX
+bluetooth: {spec: 4.0 + LE, profiles: [A2DP + aptX]}
 cameras:
 - {flash: 'LED', info: '8 MP'}
 - {flash: '', info: '1.3 MP'}
diff --git a/_data/devices/wt88047.yml b/_data/devices/wt88047.yml
index a3c5375..69cffb9 100644
--- a/_data/devices/wt88047.yml
+++ b/_data/devices/wt88047.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Removable Li-Po 2200 mAh
-bluetooth: v4.0 with A2DP, LE
+bluetooth: {spec: 4.0 + LE, profiles: [A2DP]}
 cameras:
 - {flash: 'LED', info: '8 MP'}
 - {flash: '', info: '2 MP'}
diff --git a/_data/devices/x2.yml b/_data/devices/x2.yml
index 0f6ffe0..096bf42 100644
--- a/_data/devices/x2.yml
+++ b/_data/devices/x2.yml
@@ -1,6 +1,6 @@
 architecture: arm64
 battery: 3100 mAh (non-removable)
-bluetooth: v4.2 with A2DP, BLE
+bluetooth: {spec: '4.2', profiles: [A2DP]}
 cameras:
 - {flash: 'Dual LED (dual tone)', info: '21 MP'}
 - {flash: '', info: '8 MP'}
diff --git a/_data/devices/xt897.yml b/_data/devices/xt897.yml
index 459ae84..9bb14ad 100644
--- a/_data/devices/xt897.yml
+++ b/_data/devices/xt897.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: 1785 mAh
-bluetooth: ' v4.0 with LE+EDR'
+bluetooth: {spec: 4.0 + LE}
 cameras:
 - {flash: 'LED', info: '8 MP'}
 - {flash: '', info: '1.3 MP'}
diff --git a/_data/devices/yuga.yml b/_data/devices/yuga.yml
index 58c2d1c..13cfd77 100644
--- a/_data/devices/yuga.yml
+++ b/_data/devices/yuga.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Non-removable 2330 mAh
-bluetooth: v4.0 with A2DP, LE
+bluetooth: {spec: 4.0 + LE, profiles: [A2DP]}
 cameras:
 - {flash: 'LED', info: '13.1MP f/2.4'}
 - {flash: '', info: '2MP'}
diff --git a/_data/devices/zl1.yml b/_data/devices/zl1.yml
index ff3da6f..096842c 100644
--- a/_data/devices/zl1.yml
+++ b/_data/devices/zl1.yml
@@ -1,6 +1,6 @@
 architecture: arm64
 battery: 4070 mAh (non-removable)
-bluetooth: v4.2 with A2DP, BLE
+bluetooth: {spec: '4.2', profiles: [A2DP]}
 cameras:
 - {flash: 'Dual LED (dual tone)', info: '16 MP'}
 - {flash: '', info: '8 MP'}
diff --git a/_includes/deviceinfo.html b/_includes/deviceinfo.html
index cbf7a9e..31c3824 100644
--- a/_includes/deviceinfo.html
+++ b/_includes/deviceinfo.html
@@ -127,7 +127,13 @@ ul, ol {
     {% if device.bluetooth %}
     <tr>
         <td class="header">Bluetooth</td>
-        <td>{{ device.bluetooth }}</td>
+        <td>
+            {% if device.bluetooth.profiles -%}
+                {{ device.bluetooth.spec }} with {{ device.bluetooth.profiles | join: ', ' }}
+            {% else -%}
+                {{ device.bluetooth.spec }}
+            {% endif -%}
+        </td>
     </tr>
     {% endif %}
     {% if device.wifi %}
diff --git a/device_sample/sample.yml b/device_sample/sample.yml
index fde373a..6cd58bc 100644
--- a/device_sample/sample.yml
+++ b/device_sample/sample.yml
@@ -1,6 +1,6 @@
 architecture: arm
 battery: Removable Li-Ion 2800 mAh
-bluetooth: 4.0 BLE
+bluetooth: {spec: 4.0 + LE}
 cameras:
 - {flash: 'LED', info: '16 MP'}
 - {flash: '', info: '2 MP'}
diff --git a/test/schema-06.yml b/test/schema-06.yml
index 6d45801..f9ce971 100644
--- a/test/schema-06.yml
+++ b/test/schema-06.yml
@@ -22,6 +22,41 @@ properties:
     - arm64
     - x86
     - x86_64
+  bluetooth:
+    "$id": "/properties/bluetooth"
+    properties:
+      profiles:
+        items:
+          description: Bluetooth profile
+          enum:
+          - A2DP
+          - A2DP + aptX
+          - A2DP + aptX HD
+          type: string
+        minItems: 1
+        type: array
+      spec:
+        description: Bluetooth specification
+        enum:
+        - '1.0'
+        - '1.0B'
+        - '1.1'
+        - '1.2'
+        - '2.0'
+        - '2.0 + EDR'
+        - '2.1'
+        - '2.1 + EDR'
+        - '3.0'
+        - '3.0 + HS'
+        - '4.0'
+        - '4.0 + LE'
+        - '4.1'
+        - '4.2'
+        - '5'
+        type: string
+    required:
+    - spec
+    type: object
   channels:
     "$id": "/properties/channels"
     items:
@@ -114,6 +149,7 @@ properties:
     - zuk
 required:
 - architecture
+- bluetooth
 - channels
 - codename
 - release
-- 
2.17.1

